---
title: "Goggans et al., INSERT JOURNAL 2022, Microbiome Analysis Code"
author: "Mallory Goggans, Cristian Quiroz-Moreno, Emma Bilbrey and Jessica Cooperstone"
output: github_document
---

# Introduction

INSERT ABSTRACT

Our final taxa used in this anaylsis:
-   included from Bacteria, Archaea, Eukaryota and viruses
-   removed Chordata, Arthropoda, Cnidaria, Porifera, Echinodermata, Streptophyta, Platyhelminthes because they are implausible in our biological system (pig gut microbiome)
-   remove genera/phyla that have more than 20 zeroes/33.33% missing values across our dataset of n=60

In the end: Our final dataset has 45 phyla and 755 genera.

### Load libraries
```{r, message = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

if (!requireNamespace("remotes", quietly = TRUE))
    install.packages("remotes")

if (!requireNamespace("devtools", quietly = TRUE))
    install.packages('devtools')

BiocManager::install("ALDEx2")
BiocManager::install("phyloseq")
remotes::install_github("cpauvert/psadd")
BiocManager::install("multtest")

library(devtools)

devtools::install_github("gauravsk/ranacapa")
```

```{r, message = FALSE}
# analysis packages
library(ALDEx2) # for univariate analysis
library(rstatix) # for ANOVA
library(vegan) # for beta and alpha diversity
library(phyloseq) # for krona plots and rarefaction curves
library(psadd) # additions to phyloseq package for microbiome analysis
library(ranacapa) # Utility Functions  for Simple Environmental Visualizations

# functionality packages
library(data.table) # for nicer transposing
library(here) # for directory management
library(knitr) # for knitting and for kable()
library(tidyverse) # for wrangling and plotting
library(readxl) # for reading Excel files
```

### Set seed

Some of our analyses include permutations, so let's set a seed so we get consistent results each time we run.

```{r}
set.seed(2021) # hoping this seed is better than 2020 :)
```

### Read in metadata

Input files can be found as supplementary information in:

-   UPDATE WITH MALLORY'S PAPER INFO

The data read in chunk below enables loading our data without any outside-of-R handling. In "Metadata" tab of Supplementary Information.

```{r}
# upload metadata
AllSamples.Metadata <- read_excel("../Goggans_etal_2021_tomato_pig_microbiome_WGS.xlsx",
                                       sheet = "TableS2.SampleMetadata")

str(AllSamples.Metadata)

# convert Pig, Diet, Time_Point, Diet_By_Time_Point to factors
# and set levels/order
AllSamples.Metadata$Pig <- as.factor(AllSamples.Metadata$Pig)
AllSamples.Metadata$Diet <- as.factor(AllSamples.Metadata$Diet)
AllSamples.Metadata$Time_Point <- factor(AllSamples.Metadata$Time_Point,
                                         levels = c("Day 0", "Day 7", "Day 14"))
AllSamples.Metadata$Diet_By_Time_Point <- 
  factor(AllSamples.Metadata$Diet_By_Time_Point,
         levels = c("Control Day 0", 
                  "Control Day 7", 
                  "Control Day 14", 
                  "Tomato Day 0", 
                  "Tomato Day 7", 
                  "Tomato Day 14"))

# check
str(AllSamples.Metadata)
```

# Genera-level annotation

Read in genera level data, annotated from MG-RAST. In "Genera" tab of Supplementary Information.

```{r}
Genus.AllSamples.Counts <- read_excel("../Goggans_etal_2021_tomato_pig_microbiome_WGS.xlsx",
                                       sheet = "TableS4.Genera")

str(Genus.AllSamples.Counts)
```

## Data filtering

### Remove inplausible phyla

These phyla are not plausibly found in a rectal swab of a pig, and were incorrectly annotated, so we are removing them.

```{r}
Genus.Counts.Filt <- Genus.AllSamples.Counts %>%
  filter(phylum != "Chordata" , phylum != "Arthropoda" , phylum != "Cnidaria" , 
         phylum != "Porifera" , phylum != "Echinodermata", phylum != "Streptophyta",
         phylum != "Platyhelminthes")
```

Transpose.

```{r}
Genus.Counts.Filt.t <- as.tibble(t(Genus.Counts.Filt))

# make genus colnames
colnames(Genus.Counts.Filt.t) <- Genus.Counts.Filt.t[6,]

# remove domain, phylum, class, order, family
GenusOnly.Counts.Filt.t <- Genus.Counts.Filt.t[7:66,]

# convert character to numeric
GenusOnly.Counts.Filt.t <- as.data.frame(apply((GenusOnly.Counts.Filt.t), 2, as.numeric))

str(GenusOnly.Counts.Filt.t[,1:5])

# add back sample names as column
GenusOnly.Counts.Filt.t <- GenusOnly.Counts.Filt.t %>%
  mutate(Sample_Name = AllSamples.Metadata$Sample_Name)

# move Sample_Name to first column
GenusOnly.Counts.Filt.t <- GenusOnly.Counts.Filt.t %>%
  relocate(Sample_Name)

kable(head(GenusOnly.Counts.Filt.t))
```

Calculate relative abundance, and bind back to metadata.

```{r}
GenusOnly.Counts.Filt.t.wtotal <- GenusOnly.Counts.Filt.t %>%
  mutate(Total.Counts = rowSums(GenusOnly.Counts.Filt.t[,2:ncol(GenusOnly.Counts.Filt.t)]))

dim(GenusOnly.Counts.Filt.t.wtotal)

# create rel abund df
RelAbund.Genus.Filt <- GenusOnly.Counts.Filt.t.wtotal[,2:896]/GenusOnly.Counts.Filt.t.wtotal$Total.Counts

# add back metadata
RelAbund.Genus.Filt <- bind_cols(AllSamples.Metadata, RelAbund.Genus.Filt)
```

### Counting missing data

The goal of these next bits of code are to understand how many missing values we have in our dataset, to set what parameters we will use for filtering.

```{r}
# how many zeros are in the column AHJD-like viruses?
sum(RelAbund.Genus.Filt$`AHJD-like viruses` == 0)  # this code works

# remove metadata   
# metadata is all character or factor, so can select only numeric columns
RelAbund.Genus.Filt.nometadata <- RelAbund.Genus.Filt %>%
  select_if(is.numeric)

# create a list with the number of zeros for each genus
counting_zeros <- sapply(RelAbund.Genus.Filt.nometadata, 
                         function(x){ (sum(x==0))})

# plot a histogram to look at missing values
counting_zeros_df <- as.data.frame(counting_zeros)

hist(counting_zeros_df$counting_zeros, 
     breaks = 61,
     main = "Histogram of Genera with Zero Relative Intensity",
     sub = "Starting at No Zeros",
     xlab = "Number of zero relative intensity values",
     ylab = "Frequency")
```

First column is no missing values, and its so big its hard to see how many missing values we actually have.

```{r}
# plot a histogram to look, but removing genera that are only missing 1 value
counting_zeros_df_missingval <- counting_zeros_df %>%
  rownames_to_column(var = "rowname") %>%
  filter(counting_zeros > 0) %>%
  column_to_rownames(var = "rowname")

# how many genera have at least one missing value?
dim(counting_zeros_df_missingval)
```

186 genera have at least one missing value.

```{r}
# histogram of number of zeros, starting at 1 zero
hist(counting_zeros_df_missingval$counting_zeros, 
     breaks = 60,
     main = "Histogram of Genera with Zero Relative Intensity",
     sub = "Starting at 1 Zero",
     xlab = "Number of zero relative intensity values",
     ylab = "Frequency")

# plot a histogram to look, but removing genera that have 20 or more zeros
counting_zeros_df_missing20ormore <- counting_zeros_df %>%
  rownames_to_column(var = "rowname") %>%
  filter(counting_zeros >= 20) %>%
  column_to_rownames(var = "rowname")

# histogram of number of zeros, starting at 20 zero
hist(counting_zeros_df_missing20ormore$counting_zeros, 
     breaks = 40,
     main = "Histogram of Genera with Zero Relative Intensity",
     sub = "Starting at 20 Zero",
     xlab = "Number of zero relative intensity values",
     ylab = "Frequency")
```

```{r}
# how many genera have 20 or more missing value?
dim(counting_zeros_df_missing20ormore)
```

There are 140 genera that have 20 or more missing values. Because 20 missing values here is 1/3 missing, we decided to use this as our cutoff.

### Filter for <33% missingness

Our decided criteria:\
Filter out genera from relative abundance table that have \> 20 zeros, or more than 33% missing data.

```{r}
# make a character vector of genera names that have > 20 zeros from the rownames in above table
zeros.20 <- c(rownames(counting_zeros_df_missing20ormore))

# filter using this list
RelAbund.Genus.Filt.zerofilt <- RelAbund.Genus.Filt %>%
  rownames_to_column(var = "rowname") %>%
  select(everything(), -all_of(zeros.20)) %>%
  column_to_rownames(var = "rowname")

RelAbund.Genus.Filt.zerofilt[1:3,1:6]

dim(RelAbund.Genus.Filt.zerofilt)
```

Our final dataset has 755 genera (because there are 5 columns of metadata).

Write final dataset genus rel abund to .csv this way we have it.

```{r, eval = FALSE}
write_csv(RelAbund.Genus.Filt.zerofilt,
          file = "Genus_RelAbund_Final_Filtered_WithMetadata.csv")
```

## Microbiome profile

Wrangling to enable collection of some summary statistics about our microbiome profile, including how many genera belong to different domains, etc.

### Wrangling
Grab names of final genera.

```{r}
# contains inplausible genera removed, but not removed for zeroes
dim(Genus.Counts.Filt)
Genus.Counts.Filt[1:5, 1:10]

# final filtered data
RelAbund.Genus.Filt.zerofilt[1:5, 1:10]

# grab colnames which have all the final genera
final_genera <- colnames(RelAbund.Genus.Filt.zerofilt)

# remove metadata colnames
final_genera <- final_genera[6:760]  

final_genera <- as.data.frame(final_genera)

# create a df with the final genera we want to keep for our analysis
final_genera <- final_genera %>%
  rename(genus = final_genera)
```

Get back domain and `inner_join()` with `final_genera` list

```{r}
# pull from full dataset the domain and genus columns
Genus.Counts.Filt.Domain.Genera <- Genus.Counts.Filt %>%
  select(domain, genus)

Genus.Counts.Filt.Domain.Genera[1:10,]

# want to join Genus.Counts.Filt.Domain.Genera with final_genera
final_genera_withdomain <- inner_join(final_genera, Genus.Counts.Filt.Domain.Genera,
                                      by = "genus")
```

### Count genera

```{r}
final_genera_withdomain %>%
  count()

final_genera_withdomain %>%
  group_by(domain) %>%
  count()
```

We have 755 total genera. We have 60 genera from Archaea, 582 from Bacteria, 89 from Eukaryota, and 23 from Viruses.

### Most prevalent genera
What are the most prevalent genera in our pigs?

```{r}
RelAbund.Genus.Filt.zerofilt[1:5, 1:10]

genera_means <- RelAbund.Genus.Filt.zerofilt %>%
  summarize_if(is.numeric, mean)

genera_means_t <- t(genera_means)
genera_means_t <- as.data.frame(genera_means_t)

genera_means_t <- genera_means_t %>%
  rename(rel_abund_genera = V1) %>%
  arrange(-rel_abund_genera)

head(genera_means_t)
```

The most prevalent genera are Prevotella (22.23% average abundance), Bacteroides (10.34%), Clostridium (8.56%), Lactobacillus (6.78%) and Eubacterium (5.16%).

## Rarefaction curves

### Create tax and OTU tables

This section uses a different package than the rest of the analysis; data and metadata need to be uploaded again and made into format friendly for package.

```{r}
# tax table
TAX_tab <- Genus.AllSamples.Counts %>% 
  select(Domain = domain, Phylum = phylum,
         Class = class, Order = order,
         Family = family, Genus = genus)

tax_names <- colnames(TAX_tab)

head(TAX_tab)
head(tax_names)

#  OTU table
OTU_tab <- Genus.AllSamples.Counts[, seq(7, 66)]

head(OTU_tab)
```

### Create metadata

Since metadata is contained in column names, we will parse them from here.

```{r}
raw_names <- colnames(OTU_tab)
names_table <- data.frame(Raw_names = raw_names)
```

First, the string will split by the middle hyphen.

```{r}
names_table <- names_table %>% 
  separate(Raw_names, into = c("Shotgun", "Type", "Day")) %>% 
  select(-Shotgun)
```

Now, since the character `GutMicrobiome` is constant over all samples, it will be removed. In the same manner, the character `Day` will be removed.

```{r}
names_table <- names_table %>% 
  mutate(Type = str_remove(string = Type, pattern = "GutMicrobiome") ) %>% 
  mutate(Type = str_remove(string = Type, pattern = "GutMicrobime") ) %>% 
  mutate(Day = str_remove(string = Day, pattern = "Day"))
head(names_table, 2)
```

Since `Pig` is in the middle of the sample type and the pig number, it will be used as separator character. And the final result is a tidy data.

```{r message=FALSE, warning=FALSE}
names_table <- names_table %>% 
  separate(col = Type, into = c("Type", "Pig"), sep = "Pig") %>% 
  mutate(Type = factor(Type), Pig = factor(Pig), Day = as.integer(Day)) %>% 
  select(Type, Day, Pig)
head(names_table)
```

### Renaming samples

Now that we have tidy data, it's better to replace long names with shorter ones. New names will be created as *Type_Pig_Day*.  We are also creating names that distinguish the 6 diet by time point groups.

```{r}
tmp_names  <- names_table %>%
  mutate(Pig = paste0("P", Pig), Day = paste0("D", Day)) %>% 
  unite("Kronas", Type:Day) %>% unite("Sample", Kronas:Pig, remove = F) %>% 
  select(-Pig)
head(tmp_names)
```

```{r}
metadata <- bind_cols(names_table, tmp_names)
head(metadata)
```


Finally, in order to use this metadata with the OTU table, which is linked by `names`, **the row names of the metadata and the column names in the OTU table must be the same**.

```{r}
rownames(names_table) <-  tmp_names$Sample
colnames(OTU_tab) <-   tmp_names$Sample
```

### Creating a phyloseq object

It's time create a *phyloseq* object that will allow us to analyze this data easier.

```{r warning=FALSE}
rownames(metadata) <- metadata$Sample
gut_microbiome_raw <- phyloseq(otu_table(OTU_tab, taxa_are_rows = T),
                               tax_table(TAX_tab),
                               sample_data(metadata))

colnames(tax_table(gut_microbiome_raw) ) <- tax_names
```

We can see that data at genus level accounts with 1085 taxas in 60 samples. But, we had developed a filtering scheme to remove very low abundance and inconsistently detected taxa, so let's merge this full list 

```{r}
gut_microbiome_raw
```

### Filtering taxa

We want to only include the taxa we ended up using in our final analysis.  We have already created an df `final_genera` which contains only the genera used in our final analysis.

```{r}
final_genera_forphyloseq <- final_genera$genus

# subset to include only final genera
gut_microbiome_clean <- subset_taxa(gut_microbiome_raw, Genus %in% final_genera_forphyloseq)

gut_microbiome_clean
```

The final phyloseq object has 871 taxonomic levels, in our cases, species since
it is the lowest taxonomic levels that the sequences were annotated.

In order to check if we have the 45 phyla, we are gonna to count the unique 
phyla names in the dataset.

```{r}
length(unique(tax_table(gut_microbiome_clean)[, "Genus"]))
```

We got our 755 genera.

### Creating rarefaction curves

```{r, fig.width = 4, fig.height = 6}
plot_rarefaction <- ranacapa::ggrare(gut_microbiome_clean, step = 60000,
                                   color = 'Type',  se = F, plot = F) 

plot_rarefaction <-  plot_rarefaction + theme_test() +
  facet_wrap("Day", scales = "free_x", ncol = 1, 
             labeller = labeller(Day = c( `0` = "Day 0" ,
                                          `7` = "Day 7",
                                          `14`= "Day 14")) ) + 
  labs(color = "Diet",
       title = "Rarefaction curves") +
  scale_color_manual(values = c( "steelblue2", "tomato2"))
  
plot_rarefaction
```

## Krona plots for exploratory analysis

The `psadd` package is able to create Krona plots with an phyloseq object. Two
Kronas will be created, per sample and per category `Day + Type`. The Krona plots
only include the final filtered taxa we used in our analysis.

```{r eval=FALSE, include=TRUE}
# Write kronas per samples
plot_krona(physeq = gut_microbiome, output = "./plots/kronas/per_sample", 
           variable = "Sample")

# Write kronas per category (Sample type + Day) i.e. Tomato_D7
plot_krona(physeq = gut_microbiome, output = "./plots/kronas/per_category/", 
           variable = "Kronas")
```

## PERMANOVA

Use PERMANOVA to conduct statistical analysis of overall microbial profile differences among groups.

### All samples, full model

Test the overall effect of `Diet`, `Time_Point` and their interaction of the overall microbiome.
ORIGINAL
```{r}
# create factors
factors_time_diet_pig <- RelAbund.Genus.Filt.zerofilt %>% select(Time_Point, Diet, Pig)

# create permutations
perm_time_diet_pig <- how(nperm = 9999)
setBlocks(perm_time_diet_pig) <- with(factors_time_diet_pig, Pig)

# run permanova
AllData.Genus.Filt.permanova <- adonis2(RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet*Time_Point,
                                        data = factors_time_diet_pig,
                                        permutations = perm_time_diet_pig,
                                        method = "bray")

AllData.Genus.Filt.permanova
```

-   Diet: p = 0.0001, significant   
-   Time\_Point: p = 0.0001, significant   
-   Diet\*Time\_Point: p = 0.3831, non-significant   

Comparison when you don't filter out for missing values

```{r}
AllData.Genus.Filt.permanova.no0filt <- adonis2(RelAbund.Genus.Filt[,-c(1:5)]~Diet*Time_Point,
                                        data = factors_time_diet_pig,
                                        permutations = perm_time_diet_pig,
                                        method = "bray")
AllData.Genus.Filt.permanova.no0filt
```

-   Diet: p = 0.0001, significant   
-   Time\_Point: p = 0.0001, significant   
-   Diet\*Time\_Point: p = 0.3750, non-significant   

Significance is the same whether you filter for missing data or not.

NEW
```{r}
set.seed(2021)
# create factors
factors_time_diet_pig_genus <- RelAbund.Genus.Filt.zerofilt %>% select(Time_Point, Diet, Pig)

# create permutations
perm_time_diet_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=factors_time_diet_pig_genus$Pig,
                                              type="free",))
# run permanova
AllData.Genus.Filt.permanova <- adonis2(RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet*Time_Point,
                                        data = factors_time_diet_pig_genus,
                                        permutations = perm_time_diet_pig_genus,
                                        method = "bray",
                                        by = "margin")

AllData.Genus.Filt.permanova
```

Interaction not significant (p=.355) so remove from model

```{r}
set.seed(2021)
# create factors
factors_time_diet_pig_genus <- RelAbund.Genus.Filt.zerofilt %>% select(Time_Point, Diet, Pig)

# create permutations
perm_time_diet_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=factors_time_diet_pig_genus$Pig, type="free",))
# run permanova
AllData.Genus.Filt.permanova <- adonis2(RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet+Time_Point,
                                        data = factors_time_diet_pig_genus,
                                        permutations = perm_time_diet_pig_genus,
                                        method = "bray",
                                        by = "margin")

AllData.Genus.Filt.permanova
```

Diet not significant p=.060 but close
Time significant p=.005

Test for homogeneity of multivariate dispersions

```{r}
dis <- vegdist(RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")
mod <- betadisper(dis, RelAbund.Genus.Filt.zerofilt$Diet)
permutest(mod)
```

```{r}
dis <- vegdist(RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")
mod <- betadisper(dis, RelAbund.Genus.Filt.zerofilt$Time)
permutest(mod)
```

MANOVA TRIAL

```{r, eval = FALSE}
a <- do.call(rbind, lapply(RelAbund.Genus.Filt.zerofilt, as.data.frame))

dep_vars <- cbind(RelAbund.Genus.Filt.zerofilt[-c(1:5)])
fit <- manova(cbind(RelAbund.Genus.Filt.zerofilt$Abiotrophia,RelAbund.Genus.Filt.zerofilt$Acaryochloris)~Diet*Time_Point + (1|Pig), data=RelAbund.Genus.Filt.zerofilt)

tidy(fit)
```

### Post Hoc PERMANOVA within Each Diet

#### Within Control Diet Only

Effect in control diet of time.

```{r}
set.seed(2021)
# filter for only control
control.RelAbund.Genus.Filt.zerofilt <- subset(RelAbund.Genus.Filt.zerofilt, Diet == "Control")

# create factors
factors_control_genera <- droplevels(control.RelAbund.Genus.Filt.zerofilt %>% select(Time_Point, Pig))

# create permutations
perm_control_genera <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=factors_control_genera$Pig, type="none",))
# run permanova
Control.ByTime.Genus.zerofilt <- adonis2(control.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                        data = factors_control_genera,
                                        permutations = perm_control_genera,
                                        method = "bray",
                                        by = "margin")

Control.ByTime.Genus.zerofilt
```

Significant effect of time (p = 0.005) within control samples.

Now do pairwise comparisons to see where the significance is coming from

##### Control T1 vs Control T2

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T2
control.T1T2.RelAbund.Genus.Filt.zerofilt <- subset(control.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 14")

# create factors
factors_control_T1T2_pig_genus <- droplevels(control.T1T2.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_control_T1T2_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T1T2_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T1T2.Genus.zerofilt.permanova <- adonis2(control.T1T2.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_control_T1T2_pig_genus,
                                                 permutations = perm_control_T1T2_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

Control.T1T2.Genus.zerofilt.permanova
```

Significant p = .030

##### Control T1 vs T3

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T3
control.T1T3.RelAbund.Genus.Filt.zerofilt <- subset(control.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 7")

# create factors
factors_control_T1T3_pig_genus <- droplevels(control.T1T3.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_control_T1T3_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T1T3_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T1T3.Genus.zerofilt.permanova <- adonis2(control.T1T3.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_control_T1T3_pig_genus,
                                                 permutations = perm_control_T1T3_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

Control.T1T3.Genus.zerofilt.permanova
```

Significant p = .010

##### Control T2 vs T3

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T3
control.T2T3.RelAbund.Genus.Filt.zerofilt <- subset(control.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 0")

# create factors
factors_control_T2T3_pig_genus <- droplevels(control.T2T3.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_control_T2T3_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T2T3_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T2T3.Genus.zerofilt.permanova <- adonis2(control.T2T3.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_control_T2T3_pig_genus,
                                                 permutations = perm_control_T2T3_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

Control.T2T3.Genus.zerofilt.permanova
```

Sig: p=.015

#### Tomato

Effect of tomato diet over time.

```{r}
set.seed(2021)
# filter for only tomato
tomato.RelAbund.Genus.Filt.zerofilt <- subset(RelAbund.Genus.Filt.zerofilt, Diet == "Tomato")

# create factors
factors_tomato_genera <- droplevels(tomato.RelAbund.Genus.Filt.zerofilt %>% select(Time_Point, Pig))

# create permutations
perm_tomato_genera <- how(within = Within(type="series", constant=TRUE),
                          plots = Plots(strata=factors_tomato_genera$Pig, type="none",))
# run permanova
Tomato.ByTime.Genus.zerofilt <- adonis2(tomato.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                        data = factors_tomato_genera,
                                        permutations = perm_tomato_genera,
                                        method = "bray",
                                        by = "margin")

Tomato.ByTime.Genus.zerofilt
```

Significant effect of time (p = 0.01) within tomato samples

Now do pairwise comparisons to see where the significance is coming from

##### Tomato T1 vs Tomato T2

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T2
tomato.T1T2.RelAbund.Genus.Filt.zerofilt <- subset(tomato.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 14")

# create factors
factors_tomato_T1T2_pig_genus <- droplevels(tomato.T1T2.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_tomato_T1T2_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_tomato_T1T2_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
tomato.T1T2.Genus.zerofilt.permanova <- adonis2(tomato.T1T2.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_tomato_T1T2_pig_genus,
                                                 permutations = perm_tomato_T1T2_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

tomato.T1T2.Genus.zerofilt.permanova
```

Not significant .090

##### tomato T1 vs T3

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T3
tomato.T1T3.RelAbund.Genus.Filt.zerofilt <- subset(tomato.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 7")

# create factors
factors_tomato_T1T3_pig_genus <- droplevels(tomato.T1T3.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_tomato_T1T3_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_tomato_T1T3_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
tomato.T1T3.Genus.zerofilt.permanova <- adonis2(tomato.T1T3.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_tomato_T1T3_pig_genus,
                                                 permutations = perm_tomato_T1T3_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

tomato.T1T3.Genus.zerofilt.permanova
```

Significant p = .150

##### tomato T2 vs T3

```{r}
set.seed(2021)
# filter data set for only samples at T1 and T3
tomato.T2T3.RelAbund.Genus.Filt.zerofilt <- subset(tomato.RelAbund.Genus.Filt.zerofilt,
                                               Time_Point != "Day 0")

# create factors
factors_tomato_T2T3_pig_genus <- droplevels(tomato.T2T3.RelAbund.Genus.Filt.zerofilt %>%
                                               select(Time_Point, Pig))

# create permutations
perm_tomato_T2T3_pig_genus <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_tomato_T2T3_pig_genus$Pig,
                                                 type = "free"))

# run PERMANOVA
tomato.T2T3.Genus.zerofilt.permanova <- adonis2(tomato.T2T3.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Time_Point,
                                                 data = factors_tomato_T2T3_pig_genus,
                                                 permutations = perm_tomato_T2T3_pig_genus, 
                                                 method = "bray",
                                                 by = "margin")

tomato.T2T3.Genus.zerofilt.permanova
```

p is non significant =.120

### Subset by time

#### Day 0

Effect of diet at day 0.

```{r}
# filter for day 0 only
d0.RelAbund.Genus.Filt.zerofilt <- subset(RelAbund.Genus.Filt.zerofilt, Time_Point == "Day 0")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day0_genera <- d0.RelAbund.Genus.Filt.zerofilt %>% 
  select(Diet)

# create permutations
perm_day0_genera <- how(nperm = 9999)

# run PERMANOVA
d0.ByTime.Genus.zerofilt <- adonis2(d0.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet,
                                         data = factors_day0_genera,
                                         permutations = perm_day0_genera,
                                         method = "bray")
d0.ByTime.Genus.zerofilt
```

Non-significant effect of Diet (p = 0.2402) at day 0.

#### Day 7

Effect of diet at day 7.

```{r}
# filter for day 7 only
d7.RelAbund.Genus.Filt.zerofilt <- subset(RelAbund.Genus.Filt.zerofilt, Time_Point == "Day 7")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day7_genera <- d7.RelAbund.Genus.Filt.zerofilt %>% 
  select(Diet)

# create permutations
perm_day7_genera <- how(nperm = 9999)

# run PERMANOVA
d7.ByTime.Genus.zerofilt <- adonis2(d7.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet,
                                         data = factors_day7_genera,
                                         permutations = perm_day7_genera,
                                         method = "bray")
d7.ByTime.Genus.zerofilt
```

Non-significant effect of Diet (p = 0.2836) at day 7.

#### Day 14

Effect of diet at day 14.

```{r}
# filter for day 14 only
d14.RelAbund.Genus.Filt.zerofilt <- subset(RelAbund.Genus.Filt.zerofilt, Time_Point == "Day 14")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day14_genera <- d14.RelAbund.Genus.Filt.zerofilt %>% 
  select(Diet)

# create permutations
perm_day14_genera <- how(nperm = 9999)

# run PERMANOVA
d14.ByTime.Genus.zerofilt <- adonis2(d14.RelAbund.Genus.Filt.zerofilt[,-c(1:5)]~Diet,
                                     data = factors_day14_genera,
                                     permutations = perm_day14_genera,
                                     method = "bray")
d14.ByTime.Genus.zerofilt
```

Significant effect of Diet (p = 0.005) at day 14.

## PCoA Beta Diversity

#### All samples

```{r}
# calculate distances
genus.filt.dist.20zeros <- vegdist(RelAbund.Genus.Filt.zerofilt[6:ncol(RelAbund.Genus.Filt.zerofilt)], 
                                   method = "bray")

# do multi-dimensional scaling (the PCoA calculations) on those distances
scale.genus.filt.20zeros <- cmdscale(genus.filt.dist.20zeros, k=2)

# make into data frame
scale.genus.filt.df.20zeros <- as.data.frame(cbind(scale.genus.filt.20zeros, 
                                                   AllSamples.Metadata))

# do PCoA again, but get eigen values
scale.genus.filt.20zeros.eig <- cmdscale(genus.filt.dist.20zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
eigs.genus.filt.20zeros <- (100* ((scale.genus.filt.20zeros.eig$eig)/(sum(scale.genus.filt.20zeros.eig$eig))))

# round the converted eigenvalues
round.eigs.genus.20zeros <- round(eigs.genus.filt.20zeros, 3)
```

All samples, one PCoA

```{r}
PCoA_genera_20zeros_allsamples <- scale.genus.filt.df.20zeros %>%
ggplot(aes(x = `1`, y = `2`, fill = Diet_By_Time_Point)) +
  geom_point(size=3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black"))+
  labs(x=paste("PC1: ", round.eigs.genus.20zeros[1], "%"), 
       y=paste("PC2: ", round.eigs.genus.20zeros[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Genus Level") 

PCoA_genera_20zeros_allsamples
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Genera_allsamples.png", 
       plot = PCoA_genera_20zeros_allsamples, 
       dpi = 800, 
       width = 10, 
       height = 8)
```

Re-level factors

```{r}
scale.genus.filt.df.20zeros <- scale.genus.filt.df.20zeros %>% 
  mutate(Time_Point = fct_relevel(Time_Point, c("Day 0", "Day 7", "Day 14")))
```

##### Facet by time point

```{r}
PCoA_genera_20zeros_facetbytime <- scale.genus.filt.df.20zeros %>%
ggplot(aes(x = `1`, y = `2`, fill = Diet_By_Time_Point)) +
  geom_hline(yintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_vline(xintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_point(size=3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_bw() +
  theme(axis.text = element_text(color = "black"),
        strip.background =element_rect(fill="white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  labs(x=paste("PC1: ", round.eigs.genus.20zeros[1], "%"), 
       y=paste("PC2: ", round.eigs.genus.20zeros[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Genera Level, Subset by Time Point") +
  facet_wrap(~Time_Point)

PCoA_genera_20zeros_facetbytime
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Genera_FacetByTimePoint.png", 
       plot = PCoA_genera_20zeros_facetbytime, 
       dpi = 800, 
       width = 10, 
       height = 6)
```

##### Facet by diet

```{r}
PCoA_genera_20zeros_facetbydiet <- scale.genus.filt.df.20zeros %>%
ggplot(aes(x = `1`, y = `2`, fill = Diet_By_Time_Point)) +
  geom_hline(yintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_vline(xintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_point(size=3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_bw() +
  theme(axis.text = element_text(color = "black"),
        strip.background =element_rect(fill="white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  labs(x=paste("PC1: ", round.eigs.genus.20zeros[1], "%"), 
       y=paste("PC2: ", round.eigs.genus.20zeros[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Genera Level, Subset by Diet") +
  facet_wrap(~Diet)

PCoA_genera_20zeros_facetbydiet
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Genera_FacetByDiet.png", 
       plot = PCoA_genera_20zeros_facetbydiet, 
       dpi = 800, 
       width = 10, 
       height = 6)
```

### Subset

Ended up not using this as part of the paper. Since the input is different here (i.e., the PCoA only has the subset data as an input) the output looks slightly different and we didn't feel this was the most accurate depction of the data. 

#### Control only

```{r}
# calculate distances
control.RelAbund.Genus.Filt.zerofilt.dist <- vegdist(control.RelAbund.Genus.Filt.zerofilt[,-c(1:5)], 
                                                     method = "bray")

# calculate to make PCoA
control.scale.genus.filt.20zeros <- cmdscale(control.RelAbund.Genus.Filt.zerofilt.dist, k=2)

# filter metadata
meta.control <- subset(AllSamples.Metadata, Diet == "Control")

# make into data frame and add metadata
control.scale.genus.filt.20zeros.df <- as.data.frame(cbind(meta.control, control.scale.genus.filt.20zeros))

# get eigenvalues
control.scale.genus.filt.20zeros.eig <- cmdscale(control.RelAbund.Genus.Filt.zerofilt.dist, k=2, eig = TRUE)
control.eigs.genus.filt.20zeros <- (100*((control.scale.genus.filt.20zeros.eig$eig)/(sum(control.scale.genus.filt.20zeros.eig$eig))))
control.round.eigs.genus.20zeros <- round(control.eigs.genus.filt.20zeros, 3)
```

Reset factor levels

```{r}
control.scale.genus.filt.20zeros.df$Time_Point <- factor(control.scale.genus.filt.20zeros.df$Time_Point, levels = c("Day 0", "Day 7", "Day 14"))
```

Plot

```{r}
control.scale.genus.filt.20zeros.df %>%
ggplot(aes(x = `1`, y = `2`, fill = Time_Point)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4"))+
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=(paste(control.round.eigs.genus.20zeros[1], "%")), 
       y=(paste(control.round.eigs.genus.20zeros[2], "%")), 
       fill = "Time Point",
       title = "Beta Diversity",
       subtitle = "Genera Level, Control Samples Only")
```

#### Tomato only

```{r}
# calculate distances
tomato.RelAbund.Genus.Filt.zerofilt.dist <- vegdist(tomato.RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")

# calculate to make PCoA
tomato.scale.genus.filt.20zeros <- cmdscale(tomato.RelAbund.Genus.Filt.zerofilt.dist, k=2)

# filter metadata
meta.tomato <- subset(AllSamples.Metadata, Diet == "Tomato")

# make into data frame and add metadata
tomato.scale.genus.filt.20zeros.df <- as.data.frame(cbind(meta.tomato, tomato.scale.genus.filt.20zeros))

# get eigenvalues
tomato.scale.genus.filt.20zeros.eig <- cmdscale(tomato.RelAbund.Genus.Filt.zerofilt.dist, k=2, eig = TRUE)
tomato.eigs.genus.filt.20zeros <- (100*((tomato.scale.genus.filt.20zeros.eig$eig)/(sum(tomato.scale.genus.filt.20zeros.eig$eig))))
tomato.round.eigs.genus.20zeros <- round(tomato.eigs.genus.filt.20zeros, 3)
```

Reset factor levels

```{r}
tomato.scale.genus.filt.20zeros.df$Time_Point <- factor(tomato.scale.genus.filt.20zeros.df$Time_Point, levels = c("Day 0", "Day 7", "Day 14"))
```

Plot

```{r}
tomato.scale.genus.filt.20zeros.df %>%
ggplot(aes(x = `1`, y = `2`, fill = Time_Point))+
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values = c("sienna1","firebrick3","tomato4"))+
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=(paste(tomato.round.eigs.genus.20zeros[1], "%")), 
       y=(paste(tomato.round.eigs.genus.20zeros[2], "%")), 
       fill = "Time Point",
       title = "Beta Diversity",
       subtitle = "Genera Level, Tomato Samples Only")
```

#### Day 0 Only

```{r}
# calculate distances
d0.RelAbund.Genus.Filt.zerofilt.dist <- vegdist(d0.RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")

# calculate to make PCoA
d0.scale.genus.filt.20zeros <- cmdscale(d0.RelAbund.Genus.Filt.zerofilt.dist, k=2)

# filter metadata
meta.day0 <- subset(AllSamples.Metadata, Time_Point == "Day 0")

# make into data frame and add metadata
d0.scale.genus.filt.20zeros.df <- as.data.frame(cbind(meta.day0, d0.scale.genus.filt.20zeros))

# get eigenvalues
d0.scale.genus.filt.20zeros.eig <- cmdscale(d0.RelAbund.Genus.Filt.zerofilt.dist, k=2, eig = TRUE)
d0.eigs.genus.filt.20zeros <- (100*((d0.scale.genus.filt.20zeros.eig$eig)/(sum(d0.scale.genus.filt.20zeros.eig$eig))))
d0.round.eigs.genus.20zeros <- round(d0.eigs.genus.filt.20zeros, 3)
```

Plot

```{r}
d0.scale.genus.filt.20zeros.df %>%
ggplot(aes( x= `1`, y = `2`, fill = Diet))+
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values = c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=(paste(d0.round.eigs.genus.20zeros[1], "%")), 
       y=(paste(d0.round.eigs.genus.20zeros[2], "%")),
       title = "Beta Diversity",
       subtitle = "Genera Level, Day 0 Only")
```

#### Day 7 Only

```{r}
# calculate distances
d7.RelAbund.Genus.Filt.zerofilt.dist <- vegdist(d7.RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")

# calculate to make PCoA
d7.scale.genus.filt.20zeros <- cmdscale(d7.RelAbund.Genus.Filt.zerofilt.dist, k=2)

# filter metadata
meta.day7 <- subset(AllSamples.Metadata, Time_Point == "Day 7")

# make into data frame and add metadata
d7.scale.genus.filt.20zeros.df <- as.data.frame(cbind(meta.day7, d7.scale.genus.filt.20zeros))

# get eigenvalues
d7.scale.genus.filt.20zeros.eig <- cmdscale(d7.RelAbund.Genus.Filt.zerofilt.dist, k=2, eig = TRUE)
d7.eigs.genus.filt.20zeros <- (100*((d7.scale.genus.filt.20zeros.eig$eig)/(sum(d7.scale.genus.filt.20zeros.eig$eig))))
d7.round.eigs.genus.20zeros <- round(d7.eigs.genus.filt.20zeros, 3)
```

Plot

```{r}
d7.scale.genus.filt.20zeros.df %>%
ggplot(aes( x= `1`, y = `2`, fill = Diet))+
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_color_manual(values = c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=(paste(d7.round.eigs.genus.20zeros[1], "%")), 
       y=(paste(d7.round.eigs.genus.20zeros[2], "%")),
       title = "Beta Diversity",
       subtitle = "Genera Level, Day 7 Only")
```

#### Day 14 Only

```{r}
# calculate distances
d14.RelAbund.Genus.Filt.zerofilt.dist <- vegdist(d14.RelAbund.Genus.Filt.zerofilt[,-c(1:5)], method = "bray")
# calculate to make PCoA
d14.scale.genus.filt.20zeros <- cmdscale(d14.RelAbund.Genus.Filt.zerofilt.dist, k=2)

# filter metadata
meta.day14 <- subset(AllSamples.Metadata, Time_Point == "Day 14")

# make into data frame and add metadata
d14.scale.genus.filt.20zeros.df <- as.data.frame(cbind(meta.day14, d14.scale.genus.filt.20zeros))

# get eigenvalues
d14.scale.genus.filt.20zeros.eig <- cmdscale(d14.RelAbund.Genus.Filt.zerofilt.dist, k=2, eig = TRUE)
d14.eigs.genus.filt.20zeros <- (100*((d14.scale.genus.filt.20zeros.eig$eig)/(sum(d14.scale.genus.filt.20zeros.eig$eig))))
d14.round.eigs.genus.20zeros <- round(d14.eigs.genus.filt.20zeros, 3)
```

Plot

```{r}
d14.scale.genus.filt.20zeros.df %>%
ggplot(aes( x= `1`, y = `2`, fill = Diet))+
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_color_manual(values = c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=(paste(d14.round.eigs.genus.20zeros[1], "%")), 
       y=(paste(d0.round.eigs.genus.20zeros[2], "%")),
       title = "Beta Diversity",
       subtitle = "Genera Level, Day 14 Only")
```

## Alpha Diversity

### Wrangling

```{r}
kable(head(RelAbund.Genus.Filt.zerofilt))
```

```{r}
# move Sample_Name to rownames, remove metadata
RelAbund.Genus.Filt.zerofilt.alphadiv <- RelAbund.Genus.Filt.zerofilt

rownames(RelAbund.Genus.Filt.zerofilt.alphadiv) <- RelAbund.Genus.Filt.zerofilt.alphadiv$Sample_Name  

RelAbund.Genus.Filt.zerofilt.alphadiv[1:5,1:8]

# remove metadata
RelAbund.Genus.Filt.zerofilt.alphadiv <- RelAbund.Genus.Filt.zerofilt.alphadiv %>%
  select(Abiotrophia:ncol(.))

RelAbund.Genus.Filt.zerofilt.alphadiv[1:5,1:5]

rownames(RelAbund.Genus.Filt.zerofilt.alphadiv)
```

### Calculate alpha diversity

```{r}
# run alpha diversity on phyla
genera.filt.div <- diversity(RelAbund.Genus.Filt.zerofilt.alphadiv, index = "shannon")

# convert to df
genera.filt.div.df <- as.data.frame(genera.filt.div)

# make column name 'shannon.phyla.filt'
colnames(genera.filt.div.df) <- "shannon.genera.filt"

head(genera.filt.div.df)
```

Combine shannon alpha diversity results with metadata

```{r}
# compile genera metadata
genera.metadata <- RelAbund.Genus.Filt.zerofilt[,1:5]

# combine with metadata
genera.filt.div.df.meta <- cbind(genera.metadata, genera.filt.div.df)

head(genera.filt.div.df.meta)
```

### Plotting

X axis by diet

```{r}
alpha.diversity.genera.bydiet <- genera.filt.div.df.meta %>%
  ggplot(aes(x = Diet, y = shannon.genera.filt, fill = Diet_By_Time_Point)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(fill = Diet_By_Time_Point), color = "black", alpha = 0.7, position=position_jitterdodge()) +
  scale_fill_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  scale_color_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, color = "black")) +
  labs(x=NULL, 
       y="Shannon diversity index", 
       title = "Alpha Diversity",
       subtitle = "Shannon Index, Genera Level", 
       fill="Diet & Time Point")

alpha.diversity.genera.bydiet
```

```{r, eval = FALSE}
ggsave("Figures/AlphaDiversityGenera_ByDiet_Boxplot.png", 
       plot = alpha.diversity.genera.bydiet, 
       dpi = 800, 
       width = 10, 
       height = 6)
```

X-axis by Day

```{r}
genera.filt.div.df.meta <- genera.filt.div.df.meta %>%
  mutate(Time_Point = fct_relevel(Time_Point, c("Day 0", "Day 7", "Day 14")))

alpha.diversity.genera.bytime <- genera.filt.div.df.meta %>%
  ggplot(aes(x = Time_Point, y = shannon.genera.filt, fill = Diet_By_Time_Point)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(fill = Diet_By_Time_Point), color = "black", alpha = 0.7, position=position_jitterdodge()) +
  scale_fill_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  scale_color_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, color = "black")) +
  labs(x=NULL, 
       y="Shannon diversity index", 
       title = "Alpha Diversity",
       subtitle = "Shannon Index, Genera Level", 
       fill="Diet & Time Point")

alpha.diversity.genera.bytime
```

```{r, eval = FALSE}
ggsave("Figures/AlphaDiversityGenera_ByTime_Boxplot.png", 
       plot = alpha.diversity.genera.bytime, 
       dpi = 800, 
       width = 7, 
       height = 5)
```

### Statistics

Repeated measures ANOVA on Shannon alpha diversity

```{r}
# must remove columns that aren't used in anova
head(genera.filt.div.df.meta) 

genera.filt.div.df.meta.foranova <- genera.filt.div.df.meta[,-c(1,5)]

head(genera.filt.div.df.meta.foranova)

genera.filt.alphadiv.anova <- 
  anova_test(data = genera.filt.div.df.meta.foranova,
             formula = shannon.genera.filt ~ Diet*Time_Point + Error(Pig/Time_Point),
             dv = shannon.genera.filt, 
             wid = Pig, 
             within = Time_Point, 
             between = Diet)

get_anova_table(genera.filt.alphadiv.anova)
```

-   Non-significant effect of diet (p = 0.106)   
-   Non-significant effect of timepoint (0.777)   
-   Non-significant interaction of diet:time point (p = 0.964).   

Check for normality

```{r}
shapiro.test(genera.filt.div.df.meta.foranova$shannon.genera.filt)
```

Normal.

No need for posthoc test since no model parameters are significant.

## ALDEx2

Quick introduction to anatomy of the aldex function 

The aldex function does every step - data transformation and statistics   
variable.name \<- aldex(reads.data, variables.vector, mc.samples=\#, test="t"/"kw", effect=T/F)   
reads.data - your reads/count data, unchanged    
variables.vector - a vector of the variables corresponding to sample groups, in SAME order as sample names (and therefore columns)   
mc.samples - here you tell the function how many Monte Carlo sampels to use with an integer (128 is typical)   
test - which test do you want, t-test and wilcoxon, or anova-like and kruskal wallace? (will always do the parametric and non-parametric) t = t-test and wilcoxon kw = anova-like and kruskal wallace   
effect - do you want to incude effect results in output?

Key to aldex outputs - taken directly from vignette

-   we.ep - Expected P value of Welch's t test
-   we.eBH - Expected Benjamini-Hochberg corrected P value of Welch's t test   
-   wi.ep - Expected P value of Wilcoxon rank test
-   wi.eBH - Expected Benjamini-Hochberg corrected P value of Wilcoxon test
-   kw.ep - Expected P value of Kruskal-Wallace test
-   kw.eBH - Expected Benjamini-Hochberg corrected P value of Kruskal-Wallace test
-   glm.ep - Expected P value of glm test
-   glm.eBH - Expected Benjamini-Hochberg corrected P value of glm test
-   rab.all - median clr value for all samples in the feature
-   rab.win.NS - median clr value for the NS group of samples
-   rab.win.S - median clr value for the S group of samples
-   dif.btw - median difference in clr values between S and NS groups
-   dif.win - median of the largest difference in clr values within S and NS groups
-   effect - median effect size: diff.btw / max(diff.win) for all instances
-   overlap - proportion of effect size that overlaps 0 (i.e. no effect)

ALDEx2 takes counts, not relative abundance.

We are using Benjamini Hochberg corrected pvalues, or `we.eBH` for t-tests (i.e., subsetting by time), and Benjamini-Hochberg corrected pvalues of the glm test `glm.eBH` for ANOVA tests (i.e., subsetting by diet)

Downloading ALDEx2

```{r, eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("ALDEx2")
```

### Wrangling

Since we use counts for ALDEx2, we need to filter our counts data to include only the genera we ended up using in our final analysis

```{r}
# this is the data set filtered to remove inplausible phyla, but still includes genera with a lot of missing values 
Genus.Counts.Filt[1:10,1:10]
dim(Genus.Counts.Filt)

# final genera list (after filtering for zeros)
final_genera[1:10,]

# how many final genera do we have?
dim(final_genera)

# join to create a df with genera in rows, samples in columns
# filtered for genera used in this analysis
genera_counts_foraldex <- inner_join(final_genera, Genus.Counts.Filt,
                                     by = "genus")

dim(genera_counts_foraldex)

# remove non-necessary metadata
genera_counts_foraldex <- genera_counts_foraldex[,-c(2:6)]

genera_counts_foraldex[1:10, 1:4]

# add genera as rownames
rownames(genera_counts_foraldex) <- genera_counts_foraldex$genus

# remove genera as column for cleaner data
genera_counts_foraldex <- genera_counts_foraldex %>%
  select(-genus)
```

### Subset by Time

#### Day 0

Look at the effect of diet at day 0.

```{r}
# subset day 0 only
Day0.Counts.Genera.filt <- genera_counts_foraldex %>% 
  select(ends_with("Day0"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day0.Counts.Genera.filt <- Day0.Counts.Genera.filt[order(colnames(Day0.Counts.Genera.filt))]

Diets.Day0.Genera <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day0.Genera
```

Run t-test

```{r, cache = TRUE}
filt.Genera.Day0.ByDiet.aldex <- aldex(Day0.Counts.Genera.filt, 
                                       Diets.Day0.Genera, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Genera.Day0.ByDiet.aldex <- 
  filt.Genera.Day0.ByDiet.aldex[order(filt.Genera.Day0.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Genera.Day0.ByDiet.aldex))
```

Create a histogram of pvalues of `we.eBH`

```{r}
hist(filt.Genera.Day0.ByDiet.aldex$we.eBH,
     breaks = 20,
     main = "Histogram of p-values on the effect of diet at day 0 on genera",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

`we.eBH` is the Benjamini-Hochberg corrected p-value, no significantly different genera at day 0.

#### Day 7

Look at the effect of diet at day 7.

```{r}
# subset day 7 only
Day7.Counts.Genera.filt <- genera_counts_foraldex %>% 
  select(ends_with("Day7"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day7.Counts.Genera.filt <- Day7.Counts.Genera.filt[order(colnames(Day7.Counts.Genera.filt))]

Diets.Day7.Genera <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day7.Genera
```

Run t-test

```{r, cache = TRUE}
filt.Genera.Day7.ByDiet.aldex <- aldex(Day7.Counts.Genera.filt, 
                                       Diets.Day7.Genera, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Genera.Day7.ByDiet.aldex <- 
  filt.Genera.Day7.ByDiet.aldex[order(filt.Genera.Day7.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Genera.Day7.ByDiet.aldex))
```

One genera was significantly different by diet at day 7 - unclassified (derived from bacteria), padj = 0.025

```{r}
hist(filt.Genera.Day7.ByDiet.aldex$we.eBH,
     breaks = 20,
     main = "Histogram of p-values on the effect of diet at day 7 on genera",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

What is the directionality of the change?

```{r}
filt.Genera.Day7.ByDiet.aldex %>%
  select(rab.win.Control, rab.win.Tomato, we.eBH) %>%
  filter(we.eBH <= 0.05)
```

Unclassified (derived from Bacteria) is higher in Tomato than Control.

#### Day 14

Look at the effect of diet on day 14.

```{r}
# subset day 14 only
Day14.Counts.Genera.filt <- genera_counts_foraldex %>% 
  select(ends_with("Day14"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day14.Counts.Genera.filt <- Day14.Counts.Genera.filt[order(colnames(Day14.Counts.Genera.filt))]

Diets.Day14.Genera <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day14.Genera
```

Run t-test

```{r, cache = TRUE}
filt.Genera.Day14.ByDiet.aldex <- aldex(Day14.Counts.Genera.filt, 
                                       Diets.Day14.Genera, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Genera.Day14.ByDiet.aldex <- 
  filt.Genera.Day14.ByDiet.aldex[order(filt.Genera.Day14.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Genera.Day14.ByDiet.aldex))
```

```{r}
hist(filt.Genera.Day14.ByDiet.aldex$we.eBH,
     breaks = 20,
     main = "Histogram of p-values on the effect of diet at day 14 on genera",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

How many significant genera are there?

```{r}
filt.Day14.Genera.aldex.sig <- filt.Genera.Day14.ByDiet.aldex[which(filt.Genera.Day14.ByDiet.aldex$we.eBH<0.05),]

length(rownames(filt.Day14.Genera.aldex.sig))
```

Which genera are they?

```{r}
sig_day14_genera_aldex2 <- as.data.frame(cbind(rownames(filt.Day14.Genera.aldex.sig),
                                 filt.Day14.Genera.aldex.sig$we.eBH))

sig_day14_genera_aldex2 <- sig_day14_genera_aldex2 %>%
  rename(Genera = V1,
         we.eBH_pvalue = V2)

sig_day14_genera_aldex2
```

* Lambda-like viruses
* Staphylococcus
* Alphatorquervirus
* unclassified (derived from Bacteria)
* Loa
* Plasmodium
* Propionibacterium
* Saccharomyces, Stenotrophomonas
* Malassezia
* Roseiflexus
* Brugia
* Strepococcus
* Vanderwaltozyma

What is the directionality of the change?

```{r}
filt.Genera.Day14.ByDiet.aldex %>%
  select(rab.win.Control, rab.win.Tomato, we.eBH) %>%
  filter(we.eBH <= 0.05)
```

All significantly different genera are higher in tomato as compared to control.


### Subset by diet

#### Control

```{r}
# subset control only samples across all time points, should be n=30
Control.Counts.Genera.filt <- genera_counts_foraldex %>% 
  select(contains("Control"))

dim(Control.Counts.Genera.filt)
```

ALDEx2 function needs a factor of variables

```{r}
# results in pigs at different time points being grouped together
Control.Counts.Genera.filt <- Control.Counts.Genera.filt[order(colnames(Control.Counts.Genera.filt))]

# then time point by "alphabetical" where 14 comes before 7
# ex, first few are Pig 10 Day 0, Pig 10 Day 14, Pig 10 Day 7, Pig 1 Day 0, Pig 1 Day 14, etc
TimePoints.Control.Genera <- as.vector(rep(c("Day0", "Day14", "Day7"), times=10))

# check and make sure it looks right
TimePoints.Control.Genera
```

More than two conditions this time, use the ANOVA-like test, Kruskal Wallis

```{r, cache = TRUE}
filt.Genera.Control.ByTime.aldex <- aldex(Control.Counts.Genera.filt, 
                                          TimePoints.Control.Genera, 
                                          mc.samples = 1000, 
                                          test = "kw", 
                                          effect = FALSE)
```

We are looking at `glm.eBH` for the BH corrected ANOVA pvalue

```{r}
filt.Genera.Control.ByTime.aldex <- 
  filt.Genera.Control.ByTime.aldex[order(filt.Genera.Control.ByTime.aldex$glm.eBH, 
                                         decreasing = FALSE),]

kable(head(filt.Genera.Control.ByTime.aldex))
```

```{r}
hist(filt.Genera.Control.ByTime.aldex$glm.eBH,
     breaks = 20,
     main = "Histogram of p-values on the effect of time within the control diet on genera",
     xlab = "Benjamini Hochberg corrected p-value (glm.eBH)")
```

How many significantly different genera are there?

```{r}
filt.Genera.Control.ByTime.aldex.sig <- 
  filt.Genera.Control.ByTime.aldex[which(filt.Genera.Control.ByTime.aldex$glm.eBH<0.05),]

length(rownames(filt.Genera.Control.ByTime.aldex.sig))
```

4 sig genera

Which genera are they?

```{r}
sig_control_genera_aldex2 <- as.data.frame(cbind(rownames(filt.Genera.Control.ByTime.aldex.sig),
                                 filt.Genera.Control.ByTime.aldex.sig$glm.eBH))

sig_control_genera_aldex2 <- sig_control_genera_aldex2 %>%
  rename(Genera = V1,
         glm.eBH_pval = V2)

sig_control_genera_aldex2
```

* Oribacterium
* Streptococcus
* Lactococcus
* Granulicatella

#### Tomato

```{r}
# subset tomato only samples across all time points, should be n=30
Tomato.Counts.Genera.filt <- genera_counts_foraldex %>% 
  select(contains("Tomato"))
```

ALDEx2 function needs a factor of variables

```{r}
# results in pigs at different time points being grouped together
Tomato.Counts.Genera.filt <- Tomato.Counts.Genera.filt[order(colnames(Tomato.Counts.Genera.filt))]

# then time point by "alphabetical" where 14 comes before 7
# ex, first few are Pig 10 Day 0, Pig 10 Day 14, Pig 10 Day 7, Pig 1 Day 0, Pig 1 Day 14, etc
TimePoints.Tomato.Genera <- as.vector(rep(c("Day0", "Day14", "Day7"), times=10))

# check and make sure it looks right
TimePoints.Tomato.Genera
```

More than two conditions this time, use the ANOVA-like test

```{r, cache = TRUE}
filt.Genera.Tomato.ByTime.aldex <- aldex(Tomato.Counts.Genera.filt, 
                                          TimePoints.Tomato.Genera, 
                                          mc.samples = 1000, 
                                          test = "kw", 
                                          effect = FALSE)
```

We are looking at `glm.eBH` for the BH corrected ANOVA pvalue

```{r}
filt.Genera.Tomato.ByTime.aldex <- 
  filt.Genera.Tomato.ByTime.aldex[order(filt.Genera.Tomato.ByTime.aldex$glm.eBH, 
                                         decreasing = FALSE),]

kable(head(filt.Genera.Tomato.ByTime.aldex))
```

```{r}
hist(filt.Genera.Tomato.ByTime.aldex$glm.eBH,
     breaks = 20,
     main = "Histogram of p-values on the effect of time within the tomato diet on genera",
     xlab = "Benjamini Hochberg corrected p-value (glm.eBH)")
```

How many significantly different genera are there?

```{r}
filt.Genera.Tomato.ByTime.aldex.sig <- 
  filt.Genera.Tomato.ByTime.aldex[which(filt.Genera.Tomato.ByTime.aldex$glm.eBH<0.05),]

length(rownames(filt.Genera.Tomato.ByTime.aldex.sig))
```

4 sig genera

Which genera are they?

```{r}
sig_tomato_genera_aldex2 <- as.data.frame(cbind(rownames(filt.Genera.Tomato.ByTime.aldex.sig),
                                 filt.Genera.Tomato.ByTime.aldex.sig$glm.eBH))

sig_tomato_genera_aldex2 <- sig_tomato_genera_aldex2 %>%
  rename(Genera = V1,
         glm.eBH_pval = V2)

sig_tomato_genera_aldex2
```

* Staphylococcus
* Alphatorquevirus
* Lambda-like viruses
* unclassified (derived from Bacteria)

Any overlap between sig differences at day 14 and by diet?

Control over time and day 14 overlap

```{r}
intersect(sig_day14_genera_aldex2$Genera, sig_control_genera_aldex2$Genera)
```

Streptococcus

Tomato over time and day 14 overlap

```{r}
intersect(sig_day14_genera_aldex2$Genera, sig_tomato_genera_aldex2$Genera)
```

* Lambda-like viruses
* Staphylococcus
* Alphatorquevirus
* unclassified (derived from Bacteria)

# Phyla-level annotation

Read in phyla level data, annotated from MG-RAST. In "Phyla" tab of Supplementary Information.

```{r}
Phyla.Counts <- read_excel("../Goggans_etal_2021_tomato_pig_microbiome_WGS.xlsx",
                                       sheet = "TableS3.Phyla")

str(Phyla.Counts)
```

## Data filtering

### Remove inplausible phyla

These phyla are not plausibly found in a rectal swab of a pig, and were incorrectly annotated, so we are removing them.

```{r}
Phyla.Counts.Filt <- Phyla.Counts %>%
  filter(phylum != "Chordata" , phylum != "Arthropoda" , phylum != "Cnidaria" , 
         phylum != "Porifera" , phylum != "Echinodermata", phylum != "Streptophyta",
         phylum != "Platyhelminthes")
```

Transpose.

```{r}
Phyla.Counts.Filt.t <- as.tibble(t(Phyla.Counts.Filt))

# make phyla colnames
colnames(Phyla.Counts.Filt.t) <- Phyla.Counts.Filt.t[2,]

# remove domain, phylum rows
Phyla.Counts.Filt.t <- Phyla.Counts.Filt.t[3:62,]

# convert character to numeric
Phyla.Counts.Filt.t <- as.data.frame(apply((Phyla.Counts.Filt.t), 2, as.numeric))

str(Phyla.Counts.Filt.t[,1:5])

# add back sample names as column
Phyla.Counts.Filt.t <- Phyla.Counts.Filt.t %>%
  mutate(Sample_Name = AllSamples.Metadata$Sample_Name)

# move Sample_Name to first column
Phyla.Counts.Filt.t <- Phyla.Counts.Filt.t %>%
  relocate(Sample_Name)

kable(head(Phyla.Counts.Filt.t))
```

Calculate relative abundance, and bind back to metadata.

```{r}
Phyla.Counts.Filt.t.wtotal <- Phyla.Counts.Filt.t %>%
  mutate(Total.Counts = rowSums(Phyla.Counts.Filt.t[,2:ncol(Phyla.Counts.Filt.t)]))

dim(Phyla.Counts.Filt.t.wtotal)

# create rel abund df
RelAbund.Phyla.Filt <- Phyla.Counts.Filt.t.wtotal[,2:54]/Phyla.Counts.Filt.t.wtotal$Total.Counts

# add back metadata
RelAbund.Phyla.Filt <- bind_cols(AllSamples.Metadata, RelAbund.Phyla.Filt)
```

### Counting missing data

```{r}
# remove metadata
RelAbund.Phyla.Filt.nometadata <- RelAbund.Phyla.Filt %>%
  select_if(is.numeric) 

# create a list with the number of zeros for each genus
counting_zeros_phyla <- sapply(RelAbund.Phyla.Filt.nometadata, function(x){ (sum(x==0))})

# plot a histogram to look
counting_zeros_phyla_df <- as.data.frame(counting_zeros_phyla)

hist(counting_zeros_phyla_df$counting_zeros_phyla, 
     breaks = 61,
     main = "Histogram of Genera with Zero Relative Intensity",
     sub = "Starting at No Zeros",
     xlab = "Number of zero relative intensity values",
     ylab = "Frequency")
```

Big first bar is many phyla which have zero missing values.

```{r}
# filter for any phyla with at least 1 missing value
counting_zeros_phyla_df_missingval <- counting_zeros_phyla_df %>%
  rownames_to_column(var = "rowname") %>%
  filter(counting_zeros_phyla > 0) %>%
  column_to_rownames(var = "rowname")

# how many genera have at least one missing value?
dim(counting_zeros_phyla_df_missingval)
```

9 phyla have at least 1 missing value.

```{r}
# histogram of number of zeros, starting at 1 zero
hist(counting_zeros_phyla_df_missingval$counting_zeros_phyla, 
     breaks = 60,
     main = "Histogram of Genera with Zero Relative Intensity",
     sub = "Starting at 1 Zero",
     xlab = "Number of zero relative intensity values",
     ylab = "Frequency")
```

```{r}
# create table of number of phyla with more than 1 missing value
counting_zeros_phyla_df_missingval
```

### Filter for <33% missingness

This would mean 33% missing values in our dataset.

```{r}
# removing phyla that have 20 or more zeros
counting_zeros_phyla_df_missing20ormore <- counting_zeros_phyla_df %>%
  rownames_to_column(var = "rowname") %>%
  filter(counting_zeros_phyla >= 20) %>%
  column_to_rownames(var = "rowname")

# how many phyla have 20 or more missing value?
dim(counting_zeros_phyla_df_missing20ormore)
```

8 phyla have more than 20 missing values.

```{r}
# make a character vector from the rownames of previous data frame containing the phyla we want to get rid of
phyla.20zeros <- c(rownames(counting_zeros_phyla_df_missing20ormore))

# use select function to select all columns EXCEPT the ones in the character vector, we want to remove those
# and add in metadata
RelAbund.Phyla.Filt.zerofilt <- RelAbund.Phyla.Filt %>%
  select(everything(), -all_of(phyla.20zeros))

# check dimensions to make sure it filtered correctly
dim(RelAbund.Phyla.Filt.zerofilt)
# removed 8, like we expected
```

Our final dataset has 45 phyla (because 5 columns are metadata).

Write final dataset genus rel abund to csv

```{r, eval = FALSE}
write_csv(RelAbund.Phyla.Filt.zerofilt,
          file = "Phyla_RelAbund_Final_Filtered_WithMetadata.csv")
```

## Microbiome profile

See "Genera" section above for rarefaction curves and kronas plots

### Wrangling

Wrangling to enable collection of some summary statistics about our microbiome profile.

Grab names of final phyla

```{r}
# contains inplausible genera removed, but not removed for zeroes
dim(Phyla.Counts.Filt)
Phyla.Counts.Filt[1:10, 1:5]

# final filtered data
RelAbund.Phyla.Filt.zerofilt[1:10, 1:5]
dim(RelAbund.Phyla.Filt.zerofilt)

# grab colnames which have all the final phyla
final_phyla <- colnames(RelAbund.Phyla.Filt.zerofilt)

final_phyla

# remove metadata colnames
final_phyla <- final_phyla[6:50]  

final_phyla <- as.data.frame(final_phyla)

final_phyla <- final_phyla %>%
  rename(phylum = final_phyla)
```

Get back domain and `inner_join` with `final_phyla` list

```{r}
# pull from full dataset the domain and genus columns
Phyla.Counts.Filt.Domain.Phyla <- Phyla.Counts.Filt %>%
  select(domain, phylum)

head(Phyla.Counts.Filt.Domain.Phyla)

# want to join Genus.Counts.Filt.Domain.Genera with final_phyla
final_phyla_withdomain <- inner_join(final_phyla, Phyla.Counts.Filt.Domain.Phyla,
                                     by = "phylum")
```

### Count phyla

```{r}
final_phyla_withdomain %>%
  count()

final_phyla_withdomain %>%
  group_by(domain) %>%
  count()
```

### Most prevalent phyla

```{r}
RelAbund.Phyla.Filt.zerofilt[1:5, 1:10]

phyla_means <- RelAbund.Phyla.Filt.zerofilt %>%
  summarize_if(is.numeric, mean)

phyla_means_t <- t(phyla_means)
phyla_means_t <- as.data.frame(phyla_means_t)

phyla_means_t %>%
  rename(rel_abund_phyla = V1) %>%
  arrange(-rel_abund_phyla)
```

The most prevalent phyla are Firmicutes (52.7% average abundance), Bacteroidetes (35.4%), Actinobacteria (4.7%), Proteobacteria (3.9%) and Fusobaceria (0.43%).

What percent of the reads are from Bacteria?

```{r}
final_phyla_bacteriaonly <- final_phyla_withdomain %>%
  filter(domain == "Bacteria")

final_phyla_bacteriaonly <- final_phyla_bacteriaonly$phylum

# select columns corresponding to bacteria
RelAbund.Phyla.Filt.zerofilt.baconly <- RelAbund.Phyla.Filt.zerofilt %>%
  select(contains(final_phyla_bacteriaonly)) 

# create rowsums
RelAbund.Phyla.Filt.zerofilt.baconly <- RelAbund.Phyla.Filt.zerofilt.baconly %>%
  mutate(rowsums = rowSums(RelAbund.Phyla.Filt.zerofilt.baconly[])) 

mean(RelAbund.Phyla.Filt.zerofilt.baconly$rowsums)
sd(RelAbund.Phyla.Filt.zerofilt.baconly$rowsums)

```

## PERMANOVA

### All samples, full model

Repeated measures, using Pig as a block and set permutations using `how()`
ORIGINAL BLOCK
```{r}

set.seed(2021)
# create factors
factors_time_diet_pig_phyla <- RelAbund.Phyla.Filt.zerofilt %>% 
  select(Time_Point, Diet, Pig)

# create permutations
perm_time_diet_pig_phyla <- how(nperm = 9999)
setBlocks(perm_time_diet_pig_phyla) <- with(factors_time_diet_pig_phyla, Pig)

# run permanova
AllData.Phyla.Filt.permanova <- adonis2(RelAbund.Phyla.Filt.zerofilt[,-c(1:5)]~Diet*Time_Point,
                                        data = factors_time_diet_pig_phyla,
                                        permutations = perm_time_diet_pig_phyla,
                                        method = "bray")

AllData.Phyla.Filt.permanova
```

-   Diet: p = 0.0150, significant   
-   Time\_Point: p = 0.0054, significant   
-   Diet\*Time\_Point: p = 0.4870, non-significant   

Interaction:

```{r}
# create factors
Pig <- as.factor(RelAbund.Phyla.Filt.zerofilt$Pig)
Diet <- as.factor(RelAbund.Phyla.Filt.zerofilt$Diet)


# create permutations
perm_time_diet_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=Pig, type="free",))
# run permanova
AllData.Phyla.Filt.permanova <- adonis2(RelAbund.Phyla.Filt.zerofilt[,-c(1:5)]~Diet*Time_Point,
                                        data = factors_time_diet_pig_phyla,
                                        permutations = perm_time_diet_pig_phyla,
                                        method = "bray",
                                        by = "margin")

AllData.Phyla.Filt.permanova
```

Interaction not significant (p=.51), so remove from model

```{r}
# create factors
Pig <- as.factor(RelAbund.Phyla.Filt.zerofilt$Pig)
Diet <- as.factor(RelAbund.Phyla.Filt.zerofilt$Diet)


# create permutations
perm_time_diet_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=Pig, type = "free"))
# run permanova
AllData.Phyla.Filt.permanova <- adonis2(RelAbund.Phyla.Filt.zerofilt[,-c(1:5)]~Diet + Time_Point,
                                        data = factors_time_diet_pig_phyla,
                                        permutations = perm_time_diet_pig_phyla,
                                        method = "bray",
                                        by = "margin")

AllData.Phyla.Filt.permanova
```

Test for homogeneity of multivariate dispersions

```{r}
dis <- vegdist(RelAbund.Phyla.Filt.zerofilt[,-c(1:5)], method = "bray")
mod <- betadisper(dis, Diet)
permutest(mod)
```
Non significant! good for our PERMANOVA test validity

### Post Hoc PERMANOVA within Time

#### Within Control Diet Only

Effect of control diet over time.

```{r}
# filter data set for only control samples
control.RelAbund.Phyla.zerofilt <- subset(RelAbund.Phyla.Filt.zerofilt, Diet == "Control")

# create factors
factors_control_pig_phyla <- droplevels(control.RelAbund.Phyla.zerofilt %>% 
  select(Time_Point, Pig))

# create permutations
perm_control_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                plots = Plots(strata=factors_control_pig_phyla$Pig, type = "free"))

# run PERMANOVA
Control.ByTime.Phyla.zerofilt.permanova <- adonis2(control.RelAbund.Phyla.zerofilt[,-c(1:5)]~Time_Point,
        data = factors_control_pig_phyla,
        permutations = perm_control_pig_phyla, 
        method = "bray",
        by = "margin")

Control.ByTime.Phyla.zerofilt.permanova
```

Significant effect of time (p = 0.005) within control samples. Beta diversity changing with time. Now the question is where is the difference coming from (ie. between which time points?)

##### Control T1 vs Control T2

```{r}
# filter data set for only samples at T1 and T2
control.T1T2.RelAbund.Phyla.zerofilt <- subset(control.RelAbund.Phyla.zerofilt, Time_Point != "Day 14")

# create factors
factors_control_T1T2_pig_phyla <- droplevels(control.T1T2.RelAbund.Phyla.zerofilt %>% 
  select(Time_Point, Pig))

# create permutations
perm_control_T1T2_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T1T2_pig_phyla$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T1T2.Phyla.zerofilt.permanova <- adonis2(control.T1T2.RelAbund.Phyla.zerofilt[,-c(1:5)]~Time_Point,
        data = factors_control_T1T2_pig_phyla,
        permutations = perm_control_T1T2_pig_phyla, 
        method = "bray",
        by = "margin")

Control.T1T2.Phyla.zerofilt.permanova
```
p=.085 so not significant between T1 and T2

##### Control T1 vs Control T3

```{r}
# filter data set for only samples at T1 and T3
control.T1T3.RelAbund.Phyla.zerofilt <- subset(control.RelAbund.Phyla.zerofilt, Time_Point != "Day 7")

# create factors
factors_control_T1T3_pig_phyla <- droplevels(control.T1T3.RelAbund.Phyla.zerofilt %>% 
  select(Time_Point, Pig))

# create permutations
perm_control_T1T3_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T1T3_pig_phyla$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T1T3.Phyla.zerofilt.permanova <- adonis2(control.T1T3.RelAbund.Phyla.zerofilt[,-c(1:5)]~Time_Point,
        data = factors_control_T1T3_pig_phyla,
        permutations = perm_control_T1T3_pig_phyla, 
        method = "bray",
        by = "margin")

Control.T1T3.Phyla.zerofilt.permanova
```

P = .02 so significant. There is a significant difference between T1 and T3 in the control diet pigs

##### Control T2 vs Control T3

```{r}
# filter data set for only samples at T2 and T3
control.T2T3.RelAbund.Phyla.zerofilt <- subset(control.RelAbund.Phyla.zerofilt, Time_Point != "Day 0")

# create factors
factors_control_T2T3_pig_phyla <- droplevels(control.T2T3.RelAbund.Phyla.zerofilt %>% 
  select(Time_Point, Pig))

# create permutations
perm_control_T2T3_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                                   plots = Plots(strata=factors_control_T2T3_pig_phyla$Pig,
                                                 type = "free"))

# run PERMANOVA
Control.T2T3.Phyla.zerofilt.permanova <- adonis2(control.T2T3.RelAbund.Phyla.zerofilt[,-c(1:5)]~Time_Point,
        data = factors_control_T2T3_pig_phyla,
        permutations = perm_control_T2T3_pig_phyla, 
        method = "bray",
        by = "margin")

Control.T2T3.Phyla.zerofilt.permanova
```

P = .315 so not significant

#### Within Tomato Diet Only

Effect of tomato diet over time.

```{r}
# filter data for only tomato samples
tomato.RelAbund.Phyla.zerofilt <- subset(RelAbund.Phyla.Filt.zerofilt, Diet == "Tomato")

# create factors
factors_tomato_pig_phyla <- tomato.RelAbund.Phyla.zerofilt %>% 
  select(Time_Point, Pig)

# create permutations
perm_tomato_pig_phyla <- how(within = Within(type="series", constant=TRUE),
                             plots = Plots(strata=factors_tomato_pig_phyla$Pig, type = "free"))

# run PERMANOVA
tomato.ByTime.Phyla.zerofilt.permanova <- adonis2(tomato.RelAbund.Phyla.zerofilt[,-c(1:5)]~Time_Point,
                                                  data = factors_tomato_pig_phyla,
                                                  permutations = perm_tomato_pig_phyla, 
                                                  method = "bray",
                                                  by = "margin")

tomato.ByTime.Phyla.zerofilt.permanova
```

Non-significant effect of time (p = 0.325) within tomato samples. So no post hoc tests necessary.

### Subset by time

#### Day 0 Only

Effect of diet at day 0.

```{r}
# filter data set for only day 0 samples
d0.RelAbund.Phyla.zerofilt <- subset(RelAbund.Phyla.Filt.zerofilt, Time_Point == "Day 0")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day0_phyla <- d0.RelAbund.Phyla.zerofilt %>% 
  select(Diet)

# create permutations
perm_day0_phyla <- how(nperm = 9999)

# run PERMANOVA
d0.Phyla.zerofilt.permanova <- adonis2(d0.RelAbund.Phyla.zerofilt[,-c(1:5)]~Diet,
                                       data = factors_day0_phyla,
                                       permutations = perm_day0_phyla, 
                                       method = "bray")

d0.Phyla.zerofilt.permanova
```

Non-significant effect of diet (p=0.376) at day 0.

#### Day 7 Only

Effect of diet at day 7.

```{r}
# filter data set for only day 7 samples
d7.RelAbund.Phyla.zerofilt <- subset(RelAbund.Phyla.Filt.zerofilt, Time_Point == "Day 7")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day7_phyla <- d7.RelAbund.Phyla.zerofilt %>% 
  select(Diet)

# create permutations
perm_day7_phyla <- how(nperm = 9999)

# run PERMANOVA
d7.Phyla.zerofilt.permanova <- adonis2(d7.RelAbund.Phyla.zerofilt[,-c(1:5)]~Diet,
                                       data = factors_day7_phyla,
                                       permutations = perm_day7_phyla, 
                                       method = "bray")

d7.Phyla.zerofilt.permanova
```

Non-significant effect of diet (p=0.4097) at day 7.

#### Day 14 Only

Effect of diet at day 14.

```{r}
# filter data set for only day 14 samples
d14.RelAbund.Phyla.zerofilt <- subset(RelAbund.Phyla.Filt.zerofilt, Time_Point == "Day 14")

# create factors
# don't need to include pig, since no repeated measures here 
# only testing Diet within a time point
factors_day14_phyla <- d14.RelAbund.Phyla.zerofilt %>% 
  select(Diet)

# create permutations
perm_day14_phyla <- how(nperm = 9999)

# run PERMANOVA
d14.Phyla.zerofilt.permanova <- adonis2(d14.RelAbund.Phyla.zerofilt[,-c(1:5)]~Diet,
                                       data = factors_day14_phyla,
                                       permutations = perm_day14_phyla, 
                                       method = "bray")

d14.Phyla.zerofilt.permanova
```

Non-significant effect of diet (p=0.256) at day 14.

## PCoA Beta Diversity

### All samples

```{r}
# calculate distances
phyla.filt.dist.zeros <- vegdist(RelAbund.Phyla.Filt.zerofilt[6:ncol(RelAbund.Phyla.Filt.zerofilt)], 
                                 method = "bray")

# do multi-dimensional scaling (the PCoA calculations) on those distances
scale.phyla.filt.zerofilt <- cmdscale(phyla.filt.dist.zeros, k=2)

# make into data frame and bind metadata
scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(scale.phyla.filt.zerofilt, AllSamples.Metadata))

# do PCoA again, but get eigen values
scale.phyla.filt.zerofilt.eig <- cmdscale(phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
eigs.phyla.filt.zerofilt <- (100*((scale.phyla.filt.zerofilt.eig$eig)/(sum(scale.phyla.filt.zerofilt.eig$eig))))

# round the converted eigenvalues
round.eigs.phyla.filt.zerofilt <- round(eigs.phyla.filt.zerofilt, 3)
```

Plot

```{r}
PCoA_phyla_20zeros_allsamples <- scale.phyla.filt.zerofilt.df %>%
ggplot(aes(x=`1`, y=`2`, fill = Diet_By_Time_Point)) +
  geom_point(size = 3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black"))+
  labs(x=paste("PC1: ", round.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level") 

PCoA_phyla_20zeros_allsamples
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Phyla_allsamples.png", 
       plot = PCoA_phyla_20zeros_allsamples, 
       dpi = 800, 
       width = 10, 
       height = 8)
```

#### Facet by time point

Re-level factors

```{r}
scale.phyla.filt.zerofilt.df <- scale.phyla.filt.zerofilt.df %>% 
  mutate(Time_Point = fct_relevel(Time_Point, c("Day 0", "Day 7", "Day 14")))
```

```{r}
PCoA_phyla_20zeros_facetbytime <- scale.phyla.filt.zerofilt.df %>%
ggplot(aes(x=`1`, y=`2`, fill = Diet_By_Time_Point)) +
  geom_hline(yintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_vline(xintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_point(size = 3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_bw() +
  theme(axis.text = element_text(color = "black"),
        strip.background =element_rect(fill="white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  labs(x=paste("PC1: ", round.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Subset by Time Point") +
  facet_wrap(~Time_Point)

PCoA_phyla_20zeros_facetbytime
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Phyla_FacetByTimePoint.png", 
       plot = PCoA_phyla_20zeros_facetbytime, 
       dpi = 800, 
       width = 10, 
       height = 6)
```

#### Facet by diet

```{r}
PCoA_phyla_20zeros_facetbydiet <- scale.phyla.filt.zerofilt.df %>%
ggplot(aes(x=`1`, y=`2`, fill = Diet_By_Time_Point)) +
  geom_hline(yintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_vline(xintercept = 0, color = "light grey", linetype = "dashed", size = 0.3) +
  geom_point(size = 3, color = "black", shape = 21, alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4")) +
  theme_bw() +
  theme(axis.text = element_text(color = "black"),
        strip.background =element_rect(fill="white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  labs(x=paste("PC1: ", round.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Diet & Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Subset by Diet") +
  facet_wrap(~Diet)

PCoA_phyla_20zeros_facetbydiet
```

```{r, eval = FALSE}
ggsave("Figures/BetaDiversity_PCoA_Phyla_FacetByDiet.png", 
       plot = PCoA_phyla_20zeros_facetbydiet, 
       dpi = 800, 
       width = 10, 
       height = 8)
```

### Subset

Ended up not using this as part of the paper. Since the input is different here (i.e., the PCoA only has the subset data as an input) the output looks slightly different. 

#### Control only

```{r}
# calculate distances
control.phyla.filt.dist.zeros <- vegdist(control.RelAbund.Phyla.zerofilt[,-c(1:5)], method = "bray")

# do PCoA calculations
control.scale.phyla.filt.zerofilt <- cmdscale(control.phyla.filt.dist.zeros, k=2)

# filter meta data
meta.control <- subset(AllSamples.Metadata, Diet == "Control")

# make pcoa table into data frame and bind metadata to it
control.scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(meta.control, control.scale.phyla.filt.zerofilt))

# do PCoA again, but get eigenvalues
control.scale.phyla.filt.zerofilt.eig <- cmdscale(control.phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
control.eigs.phyla.filt.zerofilt <- (100*((control.scale.phyla.filt.zerofilt.eig$eig)/sum(control.scale.phyla.filt.zerofilt.eig$eig)))

# round the eigenvalues
round.control.eigs.phyla.filt.zerofilt <- round(control.eigs.phyla.filt.zerofilt, 3)
```

Re-level factors

```{r}
control.scale.phyla.filt.zerofilt.df$Time_Point <- factor(control.scale.phyla.filt.zerofilt.df$Time_Point, 
                                                          levels = c("Day 0", "Day 7", "Day 14"))
```

Plot

```{r}
control.scale.phyla.filt.zerofilt.df %>%
ggplot(aes(x=`1`, y=`2`, fill = Time_Point)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=paste("PC1: ", round.control.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.control.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Control Only")
```

#### Tomato only

```{r}
# calculate distances
tomato.phyla.filt.dist.zeros <- vegdist(tomato.RelAbund.Phyla.zerofilt[,-c(1:5)], method = "bray")

# do PCoA calculations
tomato.scale.phyla.filt.zerofilt <- cmdscale(tomato.phyla.filt.dist.zeros, k=2)

# filter meta data
meta.tomato <- subset(AllSamples.Metadata, Diet == "Tomato")

# make pcoa table into data frame and bind metadata to it
tomato.scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(meta.tomato, tomato.scale.phyla.filt.zerofilt))

# do PCoA again, but get eigenvalues
tomato.scale.phyla.filt.zerofilt.eig <- cmdscale(tomato.phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
tomato.eigs.phyla.filt.zerofilt <- (100*((tomato.scale.phyla.filt.zerofilt.eig$eig)/sum(tomato.scale.phyla.filt.zerofilt.eig$eig)))

# round the eigenvalues
round.tomato.eigs.phyla.filt.zerofilt <- round(tomato.eigs.phyla.filt.zerofilt, 3)
```

Re-level factors

```{r}
tomato.scale.phyla.filt.zerofilt.df$Time_Point <- factor(tomato.scale.phyla.filt.zerofilt.df$Time_Point, 
                                                         levels = c("Day 0", "Day 7", "Day 14"))
```

Plot

```{r}
tomato.scale.phyla.filt.zerofilt.df %>%
ggplot(aes(x=`1`, y=`2`, fill = Time_Point)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("sienna1","firebrick3","tomato4")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=paste("PC1: ", round.tomato.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.tomato.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Tomato Only")
```

#### Day 0 Only

```{r}
# calculate distances
d0.phyla.filt.dist.zeros <- vegdist(d0.RelAbund.Phyla.zerofilt[,-c(1:5)], method = "bray")

# do PCoA calculations
d0.scale.phyla.filt.zerofilt <- cmdscale(d0.phyla.filt.dist.zeros, k=2)

# filter meta data
meta.d0 <- subset(AllSamples.Metadata, Time_Point == "Day 0")

# make pcoa table into data frame and bind metadata to it
d0.scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(meta.d0, d0.scale.phyla.filt.zerofilt))

# do PCoA again, but get eigenvalues
d0.scale.phyla.filt.zerofilt.eig <- cmdscale(d0.phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
d0.eigs.phyla.filt.zerofilt <- (100*((d0.scale.phyla.filt.zerofilt.eig$eig)/sum(d0.scale.phyla.filt.zerofilt.eig$eig)))

# round the eigenvalues
round.d0.eigs.phyla.filt.zerofilt <- round(d0.eigs.phyla.filt.zerofilt, 3)
```

Plot

```{r}
d0.scale.phyla.filt.zerofilt.df %>%
  ggplot(aes(x = `1`, y = `2`, fill = Diet)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=paste("PC1: ", round.d0.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.d0.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Day 0 Only")
```

#### Day 7 Only

```{r}
# calculate distances
d7.phyla.filt.dist.zeros <- vegdist(d7.RelAbund.Phyla.zerofilt[,-c(1:5)], method = "bray")

# do PCoA calculations
d7.scale.phyla.filt.zerofilt <- cmdscale(d7.phyla.filt.dist.zeros, k=2)

# filter meta data
meta.d7 <- subset(AllSamples.Metadata, Time_Point == "Day 7")

# make pcoa table into data frame and bind metadata to it
d7.scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(meta.d7, d7.scale.phyla.filt.zerofilt))

# do PCoA again, but get eigenvalues
d7.scale.phyla.filt.zerofilt.eig <- cmdscale(d7.phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
d7.eigs.phyla.filt.zerofilt <- (100*((d7.scale.phyla.filt.zerofilt.eig$eig)/sum(d7.scale.phyla.filt.zerofilt.eig$eig)))

# round the eigenvalues
round.d7.eigs.phyla.filt.zerofilt <- round(d7.eigs.phyla.filt.zerofilt, 3)
```

Plot

```{r}
d7.scale.phyla.filt.zerofilt.df %>%
  ggplot(aes(x = `1`, y = `2`, fill = Diet)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=paste("PC1: ", round.d7.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.d7.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Day 7 Only")
```

#### Day 14 Only

```{r}
# calculate distances
d14.phyla.filt.dist.zeros <- vegdist(d14.RelAbund.Phyla.zerofilt[,-c(1:5)], method = "bray")
# do PCoA calculations
d14.scale.phyla.filt.zerofilt <- cmdscale(d14.phyla.filt.dist.zeros, k=2)

# filter meta data
meta.d14 <- subset(AllSamples.Metadata, Time_Point == "Day 14")

# make pcoa table into data frame and bind metadata to it
d14.scale.phyla.filt.zerofilt.df <- as.data.frame(cbind(meta.d14, d14.scale.phyla.filt.zerofilt))

# do PCoA again, but get eigenvalues
d14.scale.phyla.filt.zerofilt.eig <- cmdscale(d14.phyla.filt.dist.zeros, k=2, eig = TRUE)

# convert eigenvalues to percentages and assign to a variable
d14.eigs.phyla.filt.zerofilt <- (100*((d14.scale.phyla.filt.zerofilt.eig$eig)/sum(d14.scale.phyla.filt.zerofilt.eig$eig)))

# round the eigenvalues
round.d14.eigs.phyla.filt.zerofilt <- round(d14.eigs.phyla.filt.zerofilt, 3)
```

Plot

```{r}
d14.scale.phyla.filt.zerofilt.df %>%
  ggplot(aes(x = `1`, y = `2`, fill = Diet)) +
  geom_point(size=3, shape = 21, color = "black", alpha = 0.9) +
  scale_fill_manual(values=c("steelblue2", "tomato2")) +
  theme_classic() +
  theme(axis.text = element_text(color = "black")) +
  labs(x=paste("PC1: ", round.d14.eigs.phyla.filt.zerofilt[1], "%"), 
       y=paste("PC2: ", round.d14.eigs.phyla.filt.zerofilt[2], "%"), 
       fill="Time Point",
       title = "Beta Diversity",
       subtitle = "Phyla Level, Day 14 Only")
```

## Bacteriodetes, Firmicutes, and B to F

Given a priori interest in the phyla Bacteriodetes and Firmicutes, we are conducted repeated measures ANOVA analysis for their changes in our samples. The ratio of Bacteriodetes to Firmicutes (B to F) is a commonly used metric for assessing the health of the microbiome, with a higher B to F being more beneficial.

### Wrangling

```{r}
dim(RelAbund.Phyla.Filt.zerofilt)
```

60 samples, and 45 phyla (5 columns are metadata).

Re-level Time\_Point

```{r}
RelAbund.Phyla.Filt.zerofilt <- RelAbund.Phyla.Filt.zerofilt %>%
  mutate(Time_Point = fct_relevel(Time_Point, c("Day 0", "Day 7", "Day 14")))

levels(RelAbund.Phyla.Filt.zerofilt$Time_Point)
```

Add column `Other_phyla` with the sum of all phyla that are not Bacteroidetes or Firmicutes

```{r}
RelAbund.Phyla.Filt.zerofilt.withother <- RelAbund.Phyla.Filt.zerofilt %>%
  mutate(Other_phyla = rowSums(select(.[6:ncol(.)], !contains(c("Bacteroidetes", "Firmicutes")))))

kable(head(RelAbund.Phyla.Filt.zerofilt.withother))
```

Add column B to F

```{r}
RelAbund.Phyla.Filt.zerofilt.withother.BtoF <- RelAbund.Phyla.Filt.zerofilt.withother %>%
  mutate(BtoF = Bacteroidetes/Firmicutes)

kable(head(RelAbund.Phyla.Filt.zerofilt.withother.BtoF))
```

Convert data from wide to long (i.e. make data [tidy](https://r4ds.had.co.nz/tidy-data.html))

```{r}
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF %>%
  pivot_longer(cols = 6:ncol(.),
               names_to = "phylum",
               values_to = "rel_abund")

RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long[1:10,]
```

### Plotting

Stacked bar chart of B, F, and all the other phyla

```{r}
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BFandOther <-
  RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long %>%
    filter(phylum %in% c("Bacteroidetes", "Firmicutes", "Other_phyla"))

head(RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BFandOther)

RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BFandOther %>%
  ggplot(aes(x=as.numeric(Pig), y=rel_abund, fill=phylum))+
  geom_col()+
  scale_fill_brewer(palette = "GnBu") +
  facet_grid(~Time_Point)+
  theme_classic()+
  labs(y="Relative Abundance", 
       fill="Phylum",
       x = "Pig") +
  theme(panel.grid = element_blank(), axis.text = element_text(color="black"),
        strip.text = element_text(color = "black", size = 14), 
        strip.background = element_blank())
```

### B to F Ratio

#### Plotting

B to F boxplot with jitter

```{r}
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BtoF <- 
  RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long %>%
  filter(phylum == "BtoF")

head(RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BtoF)

BtoF_Boxplot <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.BtoF %>%
  ggplot(aes(x=Diet, y=rel_abund, fill=Diet_By_Time_Point))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(aes(fill = Diet_By_Time_Point), color = "black", alpha = 0.7, position=position_jitterdodge()) +
  scale_fill_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  scale_color_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  ylim(0, 1) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, color = "black"), 
        axis.text.y = element_text(color = "black"), 
        panel.grid.minor = element_blank()) +
  labs(x=NULL, 
       y= "Bacteroidetes to Firmicutes", 
       fill="Diet & Time Point",
       title = "Ratio of Bacteroidetes to Firmicutes") 

BtoF_Boxplot
```

Saving

```{r, eval = FALSE}
ggsave("Figures/BtoFRatio_Boxplot.png", 
       plot = BtoF_Boxplot, 
       dpi = 800, 
       width = 7, 
       height = 5)
```

#### Statistics

```{r}
head(RelAbund.Phyla.Filt.zerofilt.withother.BtoF)

# select only columns used for ANOVA
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.ForANOVA <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF %>%
  select(Pig, Diet, Time_Point, BtoF)
```

Repeated measures ANOVA of B to F ratio

```{r}
BtoF.Ratio.ANOVA <- anova_test(data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF.ForANOVA, 
                          formula = BtoF ~ Diet*Time_Point + Error(Pig/(Time_Point)),
                          dv = BtoF, wid = Pig, between = Diet, within = Time_Point)

get_anova_table(BtoF.Ratio.ANOVA)
```

* Significant effect of time point (p = 0.009)   
* Nonsignificant effect of diet (p = 0.728)   
* Nonsignificant effect of diet:timepoint (p = 0.436)   

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, grouping by time (both diets)

```{r}
BtoF.Ratio.ANOVA.posthoc <- pairwise_t_test(BtoF ~ Time_Point, 
                                 data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF, 
                                 paired = TRUE, 
                                 p.adjust.method = "fdr") 

BtoF.Ratio.ANOVA.posthoc
```

Significant difference is between day 0 and day 14 (padj = 0.016).

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, separating by diet

```{r}
BtoF.Ratio.ANOVA.posthoc.bytime <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF.ForANOVA %>%
  group_by(Diet) %>%
  pairwise_t_test(BtoF ~ Time_Point, 
                  paired = TRUE, 
                  p.adjust.method = "fdr")

BtoF.Ratio.ANOVA.posthoc.bytime
```

Significant in control between day 0 and 14, padj = 0.033. All else non-significant.

### Bacteroidetes and Firmicutes

#### Plotting

Boxplotting

```{r}
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.OnlyBandF <- 
  RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long %>%
    filter(phylum == "Bacteroidetes" | phylum == "Firmicutes")

BandF_Boxplot <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF.long.OnlyBandF %>%
  ggplot(aes(x=Diet, y=rel_abund, fill=Diet_By_Time_Point))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(aes(fill = Diet_By_Time_Point), color = "black", position=position_jitterdodge(), alpha = 0.7) +
  scale_fill_manual(values=c("skyblue1", "dodgerblue", "royalblue4", "sienna1","firebrick3","tomato4"))+
  ylim(0, 0.75) +
  theme_bw() +
  facet_wrap("phylum")+
  labs(x=NULL, y= "Relative Abundance", fill="Diet & Time Point") +
  theme(axis.text.x = element_text(size = 11, color = "black"), 
        axis.text.y = element_text(color = "black"), 
        panel.grid.minor = element_blank(), 
        strip.text.x = element_text(color = "black", size = 14),
        strip.background = element_rect(fill = "white"))

BandF_Boxplot
```

Saving

```{r, eval = FALSE}
ggsave("Figures/BacteroidetesAndFirmicutes_Boxplot.png", 
       plot = BandF_Boxplot, 
       dpi = 800, 
       width = 7, 
       height = 5)
```

#### Statistics

##### Bacteroidetes

```{r}
# select only columns used for ANOVA
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Bonly <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF %>%
  select(Pig, Diet, Time_Point, Bacteroidetes)
```

Bacteroidetes repeated measures ANOVA

```{r}
Bacteroidetes.ANOVA <- anova_test(data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Bonly, 
                          formula = Bacteroidetes ~ Diet*Time_Point + Error(Pig/(Time_Point)),
                          dv = Bacteroidetes, wid = Pig, between = Diet, within = Time_Point)

get_anova_table(Bacteroidetes.ANOVA)
```

* Significant effect of time point (p = 0.024)   
* Nonsignificant effect of diet (p = 0.928)   
* onsignificant effect of diet:timepoint (p = 0.503)   

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, grouping by time (both diets)

```{r}
Bacteroidetes.ANOVA.posthoc <- pairwise_t_test(Bacteroidetes ~ Time_Point, 
                                 data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF, 
                                 paired = TRUE, 
                                 p.adjust.method = "fdr") 

Bacteroidetes.ANOVA.posthoc
```

Borderline significant difference is between day 0 and day 14 (padj = 0.058).

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, separating by diet

```{r}
Bacteroidetes.ANOVA.posthoc.bytime <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Bonly %>%
  group_by(Diet) %>%
  pairwise_t_test(Bacteroidetes ~ Time_Point, 
                  paired = TRUE, 
                  p.adjust.method = "fdr")

Bacteroidetes.ANOVA.posthoc.bytime
```

Significant difference in control between day 0 and 14, padj = 0.044. All else non-significant.

##### Firmicutes

```{r}
# select only columns used for ANOVA
RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Fonly <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF %>%
  select(Pig, Diet, Time_Point, Firmicutes)
```

Firmicutes repeated measures ANOVA

```{r}
Firmicutes.ANOVA <- anova_test(data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Fonly, 
                          formula = Firmicutes ~ Diet*Time_Point + Error(Pig/(Time_Point)),
                          dv = Firmicutes, wid = Pig, between = Diet, within = Time_Point)

get_anova_table(Firmicutes.ANOVA)
```

* Significant effect of time point (p = 0.001)   
* Nonsignificant effect of diet (p = 0.313)   
* Nonsignificant effect of diet:timepoint (p = 0.380)   

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, grouping by time (both diets)

```{r}
Firmicutes.ANOVA.posthoc <- pairwise_t_test(Firmicutes ~ Time_Point, 
                                 data = RelAbund.Phyla.Filt.zerofilt.withother.BtoF, 
                                 paired = TRUE, 
                                 p.adjust.method = "fdr") 

Firmicutes.ANOVA.posthoc
```

Significant difference is between day 0 and day 14 (padj = 0.003).

Use posthoc to see where is significant using a fdr p-value adjustment for multiple testing, separating by diet

```{r}
Firmicutes.ANOVA.posthoc.bytime <- RelAbund.Phyla.Filt.zerofilt.withother.BtoF.Fonly %>%
  group_by(Diet) %>%
  pairwise_t_test(Firmicutes ~ Time_Point, 
                  paired = TRUE, 
                  p.adjust.method = "fdr")

Firmicutes.ANOVA.posthoc.bytime
```

Significant difference is in control between day 0 and 14, padj = 0.030. All else non-significant.

## Alpha Diversity

Calculated alpha diversity of phyla based on relative abundance, including all the filtering for implausible phyla and removing samples with more than 33.33% missing samples

### Wrangling

```{r}
dim(RelAbund.Phyla.Filt.zerofilt)

RelAbund.Phyla.Filt.zerofilt[1:5,1:10]
```

Wrangle

```{r}
# move Sample_Name to rownames, remove metadata
RelAbund.Phyla.Filt.zerofilt.alphadiv <- RelAbund.Phyla.Filt.zerofilt

rownames(RelAbund.Phyla.Filt.zerofilt.alphadiv) <- RelAbund.Phyla.Filt.zerofilt.alphadiv$Sample_Name  

# remove metadata
RelAbund.Phyla.Filt.zerofilt.alphadiv <- RelAbund.Phyla.Filt.zerofilt.alphadiv %>%
  select(Acidobacteria:ncol(.))

RelAbund.Phyla.Filt.zerofilt.alphadiv[1:5,1:5]
```

### Calculate alpha diversity

```{r}
# run alpha diversity on phyla
phyla.filt.div <- diversity(RelAbund.Phyla.Filt.zerofilt.alphadiv, index = "shannon")

# convert to df
phyla.filt.div.df <- as.data.frame(phyla.filt.div)

# make column name 'shannon.phyla.filt'
colnames(phyla.filt.div.df) <- "shannon.phyla.filt"

head(phyla.filt.div.df)
```

Combine with metadata

```{r}
# combine with metadata
phyla.filt.div.df.meta <- cbind(RelAbund.Phyla.Filt.zerofilt[,1:5], phyla.filt.div.df)

head(phyla.filt.div.df.meta)
```

### Plotting

X-axis by diet

```{r}
alpha.diversity.phyla.bydiet <- phyla.filt.div.df.meta %>%
  ggplot(aes(x = Diet, y = shannon.phyla.filt, fill = Diet_By_Time_Point)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(fill = Diet_By_Time_Point), 
             color = "black", 
             alpha = 0.7, 
             position=position_jitterdodge()) +
  scale_fill_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, color = "black")) +
  labs(x=NULL, 
       y="Shannon diversity index", 
       title = "Alpha Diversity",
       subtitle = "Shannon Index, Phyla Level", 
       fill="Diet & Time Point")

alpha.diversity.phyla.bydiet
```

```{r, eval = FALSE}
ggsave("Figures/AlphaDiversityPhyla_ByDiet_Boxplot.png", 
       plot = alpha.diversity.phyla.bydiet, 
       dpi = 800, 
       width = 10, 
       height = 6)
```

X-axis by time point

```{r}
alpha.diversity.phyla.bytime <- phyla.filt.div.df.meta %>%
  ggplot(aes(x = Time_Point, y = shannon.phyla.filt, fill = Diet_By_Time_Point)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(color = "black", alpha = 0.7, position=position_jitterdodge()) +
  scale_fill_manual(values = c("skyblue1", "dodgerblue", "royalblue4", 
                               "sienna1","firebrick3","tomato4")) +
  theme_minimal() +
  theme(axis.text.x = element_text(size = 12, color = "black")) +
  labs(x=NULL, 
       y="Shannon diversity index", 
       title = "Alpha Diversity",
       subtitle = "Shannon Index, Phyla Level", 
       fill="Diet & Time Point")

alpha.diversity.phyla.bytime
```

```{r, eval = FALSE}
ggsave("Figures/AlphaDiversityPhyla_ByTime_Boxplot.png", 
       plot = alpha.diversity.phyla.bytime, 
       dpi = 800, 
       width = 7, 
       height = 5)
```

### Statistics

Repeated measures ANOVA

```{r}
# must remove columns that aren't used in anova
head(phyla.filt.div.df.meta) 

phyla.filt.div.foranova <- phyla.filt.div.df.meta[,-c(1,5)]

head(phyla.filt.div.foranova)

phyla.filt.alphadiv.anova <- 
  anova_test(data = phyla.filt.div.foranova,
             formula = shannon.phyla.filt ~ Diet*Time_Point + Error(Pig/Time_Point),
             dv = shannon.phyla.filt, 
             wid = Pig, 
             within = Time_Point, 
             between = Diet)

get_anova_table(phyla.filt.alphadiv.anova)
```

* Significant effect of diet (p = 0.004)   
* Non-significant effect of timepoint (0.086)   
* Non-significant interaction of diet:time point (p = 0.791).   

Check for normality

```{r}
shapiro.test(phyla.filt.div.df.meta$shannon.phyla.filt)
```

Normal.

Post-hoc tests

```{r}
posthoc.morevariables <- phyla.filt.div.foranova %>%
  group_by(Time_Point) %>%
  anova_test(dv = shannon.phyla.filt, wid = Pig, between = Diet) %>%
  get_anova_table() %>%
  adjust_pvalue(method = "fdr")

posthoc.morevariables
```

Significant effect of diet at day 14 (padj = 0.024)

```{r}
posthoc.evenmorespecific <- phyla.filt.div.foranova %>%
  group_by(Time_Point) %>%
  pairwise_t_test(shannon.phyla.filt ~ Diet,
                  paired = TRUE,
                  p.adjust.method = "fdr")

posthoc.evenmorespecific
```

* Significant effect between control and tomato at day 14 (padj = 0.01)    
* Nonsignificant at day 0 (padj = 0.328)   
* Nonsignificant but getting close at day 7 (padj = 0.085)   

## ALDEx2

Quick introduction to anatomy of the aldex function 

The aldex function does every step - data transformation and statistics    
variable.name \<- aldex(reads.data, variables.vector, mc.samples=\#, test="t"/"kw", effect=T/F)   
reads.data - your reads/count data, un changed   
variables.vector - a vector of the variables corresponding to sample groups, in SAME order as sample names (and therefore columns)    
mc.samples - here you tell the function how many Monte Carlo sampels to use with an integer (128 is typical)    
test - which test do you want, t-test and wilcoxon, or anova-like and kruskal wallace? (will always do the parametric and non-parametric) t = t-test and wilcoxon kw = anova-like and kruskal wallace    
effect - do you want it to incude effect results in output?

Key to aldex outputs - taken directly from vignette

-   we.ep - Expected P value of Welch's t test
-   we.eBH - Expected Benjamini-Hochberg corrected P value of Welch's t test   
-   wi.ep - Expected P value of Wilcoxon rank test
-   wi.eBH - Expected Benjamini-Hochberg corrected P value of Wilcoxon test
-   kw.ep - Expected P value of Kruskal-Wallace test
-   kw.eBH - Expected Benjamini-Hochberg corrected P value of Kruskal-Wallace test
-   glm.ep - Expected P value of glm test
-   glm.eBH - Expected Benjamini-Hochberg corrected P value of glm test
-   rab.all - median clr value for all samples in the feature
-   rab.win.NS - median clr value for the NS group of samples
-   rab.win.S - median clr value for the S group of samples
-   dif.btw - median difference in clr values between S and NS groups
-   dif.win - median of the largest difference in clr values within S and NS groups
-   effect - median effect size: diff.btw / max(diff.win) for all instances
-   overlap - proportion of effect size that overlaps 0 (i.e. no effect)

ALDEx2 takes counts, not relative abundance.

We are using Benjamini Hochberg corrected pvalues, or `we.eBH` for t-tests (i.e., subsetting by time), and Benjamini-Hochberg corrected pvalues of the glm test `glm.eBH` for ANOVA tests (i.e., subsetting by diet)

Downloading ALDEx2

```{r, eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("ALDEx2")
```

### Wrangling

Since we use counts for ALDEx2, we need to filter our counts data to include only the phyla we ended up using in our final analysis

```{r}
# this data set filtered to remove inplausible phyla, but still includes phyla with a lot of missing values 
Phyla.Counts.Filt[1:10,1:10]
dim(Phyla.Counts.Filt)

# final phyla list, after filtering for number of zeros
final_phyla[1:10,]

# how many final phyla do we have?
dim(final_phyla)

# join to create a df with phyla in rows, samples in columns
# filtered for genera used in this analysis
phyla_counts_foraldex <- inner_join(final_phyla, Phyla.Counts.Filt,
                                    by = "phylum")

dim(phyla_counts_foraldex)
phyla_counts_foraldex[1:10, 1:4]

# add phyla as rownames
rownames(phyla_counts_foraldex) <- phyla_counts_foraldex$phylum

# remove phylum, domain as columns for cleaner data
phyla_counts_foraldex <- phyla_counts_foraldex %>%
  select(-phylum, -domain)
```

### Subset by Time

#### Day 0

```{r}
# subset day 0 only
Day0.Counts.Phyla.filt <- phyla_counts_foraldex %>% 
  select(ends_with("Day0"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day0.Counts.Phyla.filt <- Day0.Counts.Phyla.filt[order(colnames(Day0.Counts.Phyla.filt))]

Diets.Day0.Phyla <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day0.Phyla
```

Run t-tests

```{r, cache = TRUE}
# runs very slowly
# set cache = TRUE to save results
filt.Phyla.Day0.ByDiet.aldex <- aldex(Day0.Counts.Phyla.filt, 
                                       Diets.Day0.Phyla, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Phyla.Day0.ByDiet.aldex <- 
  filt.Phyla.Day0.ByDiet.aldex[order(filt.Phyla.Day0.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Phyla.Day0.ByDiet.aldex))
```

No significantly different phyla

```{r}
hist(filt.Phyla.Day0.ByDiet.aldex$we.eBH,
     breaks = 45,
     main = "Histogram of p-values on the effect of diet at day 0 on phyla",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

`we.eBH` is the Benjamini-Hochberg corrected p-value, and nothing is \< 0.05

#### Day 7

```{r}
# subset day 7 only
Day7.Counts.Phyla.filt <- phyla_counts_foraldex %>% 
  select(ends_with("Day7"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day7.Counts.Phyla.filt <- Day7.Counts.Phyla.filt[order(colnames(Day7.Counts.Phyla.filt))]

Diets.Day7.Phyla <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day7.Phyla
```

Run t-tests

```{r, cache = TRUE}
# runs very slowly
# set cache = TRUE to save results
filt.Phyla.Day7.ByDiet.aldex <- aldex(Day7.Counts.Phyla.filt, 
                                       Diets.Day7.Phyla, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Phyla.Day7.ByDiet.aldex <- 
  filt.Phyla.Day7.ByDiet.aldex[order(filt.Phyla.Day7.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Phyla.Day7.ByDiet.aldex))
```

One phyla was significantly different by diet at day 7 - unclassified (derived from bacteria), padj = 0.002

```{r}
hist(filt.Phyla.Day7.ByDiet.aldex$we.eBH,
     breaks = 45,
     main = "Histogram of p-values on the effect of diet at day 7 on phyla",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

#### Day 14

```{r}
# subset day 14 only
Day14.Counts.Phyla.filt <- phyla_counts_foraldex %>% 
  select(ends_with("Day14"))
```

ALDEx2 function needs a factor of variables

```{r}
# order alphabetically so making the meta data vector is easier
Day14.Counts.Phyla.filt <- Day14.Counts.Phyla.filt[order(colnames(Day14.Counts.Phyla.filt))]

Diets.Day14.Phyla <- as.vector(c(rep("Control", times=10), rep("Tomato", times=10)))

# check and make sure it came out right
Diets.Day14.Phyla
```

Run t-tests

```{r, cache = TRUE}
filt.Phyla.Day14.ByDiet.aldex <- aldex(Day14.Counts.Phyla.filt, 
                                       Diets.Day14.Phyla, 
                                       mc.samples = 1000, 
                                       test = "t", 
                                       effect = TRUE)
```

```{r}
filt.Phyla.Day14.ByDiet.aldex <- 
  filt.Phyla.Day14.ByDiet.aldex[order(filt.Phyla.Day14.ByDiet.aldex$we.eBH, 
                                      decreasing = FALSE),]

kable(head(filt.Phyla.Day14.ByDiet.aldex))
```

```{r}
hist(filt.Phyla.Day14.ByDiet.aldex$we.eBH,
     breaks = 45,
     main = "Histogram of p-values on the effect of diet at day 14 on phyla",
     xlab = "Benjamini Hochberg corrected p-value (we.eBH)")
```

How many significant phyla are there?

```{r}
filt.Phyla.Day14.ByDiet.aldex.sig <- 
  filt.Phyla.Day14.ByDiet.aldex[which(filt.Phyla.Day14.ByDiet.aldex$we.eBH<0.05),]

length(rownames(filt.Phyla.Day14.ByDiet.aldex.sig))
```

5 sig phyla

Which phyla are they?

```{r}
sig_day14_phyla_aldex2 <- as.data.frame(cbind(rownames(filt.Phyla.Day14.ByDiet.aldex.sig),
                                 filt.Phyla.Day14.ByDiet.aldex.sig$we.eBH))

sig_day14_phyla_aldex2
```

* unclassified (derived from Bacteria)
* Nematoda
* Apicomplexa
* Deinococcus-Thermus
* Proteobacteria

What is the directionality of the change?

```{r}
filt.Phyla.Day14.ByDiet.aldex %>%
  select(rab.win.Control, rab.win.Tomato, we.eBH) %>%
  filter(we.eBH <= 0.05)
```

Higher in control:   
* Deinococcus-Thermus

Higher in tomato:   
* unclassified (derived from Bacteria)   
* Nematoda   
* Apicomplexa   
* Proteobacteria   

### Subset by diet

#### Control

```{r}
# subset control only samples across all time points, should be n=30
Control.Counts.Phyla.filt <- phyla_counts_foraldex %>% 
  select(contains("Control"))
```

ALDEx2 function needs a factor of variables

```{r}
# results in pigs at different time points being grouped together
Control.Counts.Phyla.filt <- Control.Counts.Phyla.filt[order(colnames(Control.Counts.Phyla.filt))]

# then time point by "alphabetical" where 14 comes before 7
# ex, first few are Pig 10 Day 0, Pig 10 Day 14, Pig 10 Day 7, Pig 1 Day 0, Pig 1 Day 14, etc

TimePoints.Control.Phyla <- as.vector(rep(c("Day0", "Day14", "Day7"), times=10))

# check and make sure it looks right
TimePoints.Control.Phyla
```

More than two conditions this time, use the ANOVA-like test

```{r, cache = TRUE}
filt.Phyla.Control.ByTime.aldex <- aldex(Control.Counts.Phyla.filt, 
                                          TimePoints.Control.Phyla, 
                                          mc.samples = 1000, 
                                          test = "kw", 
                                          effect = FALSE)
```

We are looking at `glm.eBH` for the BH corrected ANOVA pval

```{r}
filt.Phyla.Control.ByTime.aldex <- 
  filt.Phyla.Control.ByTime.aldex[order(filt.Phyla.Control.ByTime.aldex$glm.eBH, 
                                         decreasing = FALSE),]

kable(head(filt.Phyla.Control.ByTime.aldex))
```

```{r}
hist(filt.Phyla.Control.ByTime.aldex$glm.eBH,
     breaks = 45,
     main = "Histogram of p-values on the effect of time on control pigs on phyla",
     xlab = "Benjamini Hochberg corrected p-value (glm.eBH)")
```

How many significant phyla are there?

```{r}
filt.Phyla.Control.ByTime.aldex.sig <- 
  filt.Phyla.Control.ByTime.aldex[which(filt.Phyla.Control.ByTime.aldex$glm.eBH<0.05),]

length(rownames(filt.Phyla.Control.ByTime.aldex.sig))
```

0 sig phyla

#### Tomato

```{r}
# subset tomato only samples across all time points, should be n=30
Tomato.Counts.Phyla.filt <- phyla_counts_foraldex %>% 
  select(contains("Tomato"))
```

ALDEx2 function needs a factor of variables

```{r}
# results in pigs at different time points being grouped together
Tomato.Counts.Phyla.filt <- Tomato.Counts.Phyla.filt[order(colnames(Tomato.Counts.Phyla.filt))]

# then time point by "alphabetical" where 14 comes before 7
# ex, first few are Pig 10 Day 0, Pig 10 Day 14, Pig 10 Day 7, Pig 1 Day 0, Pig 1 Day 14, etc

TimePoints.Tomato.Phyla <- as.vector(rep(c("Day0", "Day14", "Day7"), times=10))

# check and make sure it looks right
TimePoints.Tomato.Phyla
```

More than two conditions this time, use the ANOVA-like test

```{r, cache = TRUE}
filt.Phyla.Tomato.ByTime.aldex <- aldex(Tomato.Counts.Phyla.filt, 
                                          TimePoints.Tomato.Phyla, 
                                          mc.samples = 1000, 
                                          test = "kw", 
                                          effect = FALSE)
```

We are looking at glm.eBH for the BH corrected ANOVA pval

```{r}
filt.Phyla.Tomato.ByTime.aldex <- 
  filt.Phyla.Tomato.ByTime.aldex[order(filt.Phyla.Tomato.ByTime.aldex$glm.eBH, 
                                         decreasing = FALSE),]

kable(head(filt.Phyla.Tomato.ByTime.aldex))
```

```{r}
hist(filt.Phyla.Tomato.ByTime.aldex$glm.eBH,
     breaks = 45,
     main = "Histogram of p-values on the effect of time on tomato pigs on phyla",
     xlab = "Benjamini Hochberg corrected p-value (glm.eBH)")
```

How many significant phyla are there?

```{r}
filt.Phyla.Tomato.ByTime.aldex.sig <- 
  filt.Phyla.Tomato.ByTime.aldex[which(filt.Phyla.Tomato.ByTime.aldex$glm.eBH<0.05),]

length(rownames(filt.Phyla.Tomato.ByTime.aldex.sig))
```

1 sig phyla, unclassified (derived from Bacteria)
